diff -urN glibc-2.6.org/intl/locale.alias glibc-2.6/intl/locale.alias
--- glibc-2.6.org/intl/locale.alias	2003-12-04 08:57:47.000000000 +0100
+++ glibc-2.6/intl/locale.alias	2007-05-18 12:14:44.909053462 +0200
@@ -49,6 +49,9 @@
 hungarian       hu_HU.ISO-8859-2
 icelandic       is_IS.ISO-8859-1
 italian         it_IT.ISO-8859-1
+iw_IL		he_IL.ISO-8859-8
+iw_IL.ISO-8859-8 he_IL.ISO-8859-8
+iw_IL.UTF-8	he_IL.UTF-8
 japanese	ja_JP.eucJP
 japanese.euc	ja_JP.eucJP
 ja_JP		ja_JP.eucJP
@@ -60,6 +63,8 @@
 lithuanian      lt_LT.ISO-8859-13
 no_NO		nb_NO.ISO-8859-1
 no_NO.ISO-8859-1 nb_NO.ISO-8859-1
+no_NO.ISO-8859-15 nb_NO.ISO-8859-15
+no_NO.UTF-8	nb_NO.UTF-8
 norwegian       nb_NO.ISO-8859-1
 nynorsk		nn_NO.ISO-8859-1
 polish          pl_PL.ISO-8859-2
@@ -70,6 +75,12 @@
 slovene         sl_SI.ISO-8859-2
 slovenian       sl_SI.ISO-8859-2
 spanish         es_ES.ISO-8859-1
+sr_RS@Latn	sr_RS.ISO-8859-2@latin
+sr_RS.ISO-8859-2@Latn sr_RS.ISO-8859-2@latin
+sr_YU		sr_RS.ISO-8859-2@latin
+sr_YU.ISO-8859-2 sr_RS.ISO-8859-2@latin
+sr_YU@cyrillic	sr_RS.ISO-8859-5
+sr_YU.ISO-8859-5@cyrillic sr_RS.ISO-8859-5
 swedish         sv_SE.ISO-8859-1
 thai		th_TH.TIS-620
 turkish         tr_TR.ISO-8859-9
diff -urN glibc-2.6.org/localedata/locales/be_BY@alternative glibc-2.6/localedata/locales/be_BY@alternative
--- glibc-2.6.org/localedata/locales/be_BY@alternative	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/be_BY@alternative	2007-05-18 12:14:53.723180198 +0200
@@ -0,0 +1,139 @@
+comment_char %
+escape_char  /
+%
+% Belarusian Language Locale for Belarus
+% Contact: Alexander Mikhailian
+% Email: mikhailian@altern.org
+% Language: be
+% Territory: BY
+% Revision: 0.5
+% Date: 2004-08-24
+% Application: general
+% Users: general
+% Repertoiremap: mnemonic.ds
+% Charset: CP1251, UTF-8
+% Distribution and use is free, also
+% for commercial purposes.
+
+LC_IDENTIFICATION
+title      "Belarusian locale for Belarus, traditional spelling"
+source     "Belarusian i18n mailing list"
+address    "i18n@mova.org"
+contact    "Alexander Mikhailian"
+email      "mikhailian@altern.org"
+tel        "+32 494 60 91 31"
+fax        ""
+language   "Belarusian"
+territory  "Belarus"
+revision   "1.0"
+date       "2000-06-29"
+audience   ""
+application ""
+abbreviation "clasicny"
+%
+category  "be_BY:2000";LC_IDENTIFICATION
+category  "be_BY:2000";LC_CTYPE
+category  "be_BY:2000";LC_COLLATE
+category  "be_BY:2000";LC_TIME
+category  "be_BY:2000";LC_NUMERIC
+category  "be_BY:2000";LC_MONETARY
+category  "be_BY:2000";LC_MESSAGES
+category  "be_BY:2000";LC_PAPER
+category  "be_BY:2000";LC_TELEPHONE
+category  "be_BY:2000";LC_MEASUREMENT
+category  "be_BY:2000";LC_NAME
+category  "be_BY:2000";LC_ADDRESS
+
+END LC_IDENTIFICATION
+
+LC_COLLATE
+copy "be_BY"
+END LC_COLLATE
+
+LC_CTYPE
+copy "i18n"
+END LC_CTYPE
+
+LC_MESSAGES
+copy "be_BY"
+END LC_MESSAGES
+
+LC_MONETARY
+copy "be_BY"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "be_BY"
+END LC_NUMERIC
+
+LC_TIME
+day     "<U041D><U044F><U0434><U0437><U0435><U043B><U044F>";/
+        "<U041F><U0430><U043D><U044F><U0434><U0437><U0435><U043B><U0430><U043A>";/
+        "<U0410><U045E><U0442><U043E><U0440><U0430><U043A>";/
+        "<U0421><U0435><U0440><U0430><U0434><U0430>";/
+        "<U0427><U0430><U0446><U044C><U0432><U0435><U0440>";/
+        "<U041F><U044F><U0442><U043D><U0456><U0446><U0430>";/
+        "<U0421><U0443><U0431><U043E><U0442><U0430>"
+abday   "<U041D><U044F><U0434>";/
+        "<U041F><U0430><U043D>";/
+        "<U0410><U045E><U0442>";/
+        "<U0421><U0440><U0434>";/
+        "<U0427><U0446><U0432>";/
+        "<U041F><U044F><U0442>";/
+        "<U0421><U0443><U0431>"
+first_weekday 2
+first_workday 2
+mon     "<U0421><U0442><U0443><U0434><U0437><U0435><U043D><U044C>";/
+        "<U041B><U044E><U0442><U044B>";/
+        "<U0421><U0430><U043A><U0430><U0432><U0456><U043A>";/
+        "<U041A><U0440><U0430><U0441><U0430><U0432><U0456><U043A>";/
+        "<U0422><U0440><U0430><U0432><U0435><U043D><U044C>";/
+        "<U0427><U044D><U0440><U0432><U0435><U043D><U044C>";/
+        "<U041B><U0456><U043F><U0435><U043D><U044C>";/
+        "<U0416><U043D><U0456><U0432><U0435><U043D><U044C>";/
+        "<U0412><U0435><U0440><U0430><U0441><U0435><U043D><U044C>";/
+        "<U041A><U0430><U0441><U0442><U0440><U044B><U0447><U043D><U0456><U043A>";/
+        "<U041B><U0456><U0441><U0442><U0430><U043F><U0430><U0434>";/
+        "<U0421><U044C><U043D><U0435><U0436><U0430><U043D><U044C>"
+abmon   "<U0421><U0442><U0434>";/
+        "<U041B><U044E><U0442>";/
+        "<U0421><U0430><U043A>";/
+        "<U041A><U0440><U0441>";/
+        "<U0422><U0440><U0430>";/
+        "<U0427><U044D><U0440>";/
+        "<U041B><U0456><U043F>";/
+        "<U0416><U043D><U0432>";/
+        "<U0412><U0440><U0441>";/
+        "<U041A><U0441><U0442>";/
+        "<U041B><U0456><U0441>";/
+        "<U0421><U043D><U0436>"
+d_t_fmt "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062>/
+<U0020><U0025><U0059><U0020><U0025><U0054>"
+d_fmt   "<U0025><U0064><U002E><U0025><U006D><U002E><U0025><U0059>"
+t_fmt   "<U0025><U0054>"
+am_pm   "";""
+t_fmt_ampm  ""
+date_fmt    "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
+<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U005A><U0020><U0025><U0059>"
+END LC_TIME
+
+LC_PAPER
+copy "be_BY"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "be_BY"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "be_BY"
+END LC_MEASUREMENT
+
+LC_NAME
+copy "be_BY"
+END LC_NAME
+
+LC_ADDRESS
+copy "be_BY"
+END LC_ADDRESS
diff -urN glibc-2.6.org/localedata/locales/eo glibc-2.6/localedata/locales/eo
--- glibc-2.6.org/localedata/locales/eo	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/eo	2007-05-18 12:14:53.726513832 +0200
@@ -0,0 +1,216 @@
+comment_char %
+escape_char /
+
+% Esperanto language locale for ...
+% Edmund GRIMLEY EVANS <edmundo at rano dot org>
+
+LC_IDENTIFICATION
+title      "Esperanto locale"
+source     ""
+address    ""
+contact    ""
+email      "debian-esperanto@lists.debian.org"
+tel        ""
+fax        ""
+language   "Esperanto"
+territory  ""
+revision   "draft"
+date       "2002-07-04"
+%
+category  "eo_XX:2000";LC_IDENTIFICATION
+category  "eo_XX:2000";LC_CTYPE
+category  "eo_XX:2000";LC_COLLATE
+category  "eo_XX:2000";LC_TIME
+category  "eo_XX:2000";LC_NUMERIC
+category  "eo_XX:2000";LC_MONETARY
+category  "eo_XX:2000";LC_MESSAGES
+category  "eo_XX:2000";LC_PAPER
+category  "eo_XX:2000";LC_NAME
+category  "eo_XX:2000";LC_ADDRESS
+category  "eo_XX:2000";LC_TELEPHONE
+
+END LC_IDENTIFICATION
+
+LC_CTYPE
+copy "i18n"
+END LC_CTYPE
+
+LC_COLLATE
+copy "iso14651_t1"
+
+collating-symbol <ccirc>
+collating-symbol <gcirc>
+collating-symbol <hcirc>
+collating-symbol <jcirc>
+collating-symbol <scirc>
+collating-symbol <ubreve>
+
+reorder-after <c>
+<ccirc>
+reorder-after <g>
+<gcirc>
+reorder-after <h>
+<hcirc>
+reorder-after <j>
+<jcirc>
+reorder-after <s>
+<scirc>
+reorder-after <u>
+<ubreve>
+
+reorder-after <U0043>
+<U0108> <ccirc>;<CIR>;<CAP>;IGNORE % Ĉ
+reorder-after <U0063>
+<U0109> <ccirc>;<CIR>;<MIN>;IGNORE % ĉ
+reorder-after <U0047>
+<U011C> <gcirc>;<CIR>;<CAP>;IGNORE % Ĝ
+reorder-after <U0067>
+<U011D> <gcirc>;<CIR>;<MIN>;IGNORE % ĝ
+reorder-after <U0048>
+<U0124> <hcirc>;<CIR>;<CAP>;IGNORE % Ĥ
+reorder-after <U0068>
+<U0125> <hcirc>;<CIR>;<MIN>;IGNORE % ĥ
+reorder-after <U004A>
+<U0134> <jcirc>;<CIR>;<CAP>;IGNORE % Ĵ
+reorder-after <U006A>
+<U0135> <jcirc>;<CIR>;<MIN>;IGNORE % ĵ
+reorder-after <U0053>
+<U015C> <scirc>;<CIR>;<CAP>;IGNORE % Ŝ
+reorder-after <U0073>
+<U015D> <scirc>;<CIR>;<MIN>;IGNORE % ŝ
+reorder-after <U0055>
+<U016C> <ubreve>;<BRE>;<CAP>;IGNORE % Ŭ
+reorder-after <U0075>
+<U016D> <ubreve>;<BRE>;<MIN>;IGNORE % ŭ
+
+reorder-end
+
+END LC_COLLATE
+
+LC_MONETARY
+int_curr_symbol         "<U0045><U0055><U0052><U0020>" % "EUR"
+currency_symbol         "<U20AC>" % "EUR"
+% "<U20AC>" ne eblas en ISO-8859-3
+mon_decimal_point       "<U002C>"
+mon_thousands_sep       "<U0020>"
+mon_grouping            3;3
+positive_sign           ""
+negative_sign           "<U002D>"
+int_frac_digits         2
+frac_digits             2
+p_cs_precedes           1
+p_sep_by_space          1
+n_cs_precedes           1
+n_sep_by_space          1
+p_sign_posn             1
+n_sign_posn             1
+END LC_MONETARY
+
+LC_NUMERIC
+decimal_point           "<U002C>"
+thousands_sep           ""
+grouping                0;0
+END LC_NUMERIC
+
+LC_TIME
+
+% "dim";"lun";"mar";"mer";"ĵaŭ";"ven";"sab"
+abday	"<U0064><U0069><U006D>";"<U006C><U0075><U006E>";/
+	"<U006D><U0061><U0072>";"<U006D><U0065><U0072>";/
+	"<U0135><U0061><U016D>";"<U0076><U0065><U006E>";/
+	"<U0073><U0061><U0062>"
+
+% "dimanĉo";"lundo";"mardo";"merkredo";"ĵaŭdo";"vendredo";"sabato"
+day	"<U0064><U0069><U006D><U0061><U006E><U0109><U006F>";/
+	"<U006C><U0075><U006E><U0064><U006F>";/
+	"<U006D><U0061><U0072><U0064><U006F>";/
+	"<U006D><U0065><U0072><U006B><U0072><U0065><U0064><U006F>";/
+	"<U0135><U0061><U016D><U0064><U006F>";/
+	"<U0076><U0065><U006E><U0064><U0072><U0065><U0064><U006F>";/
+	"<U0073><U0061><U0062><U0061><U0074><U006F>"
+
+% "Jan";"Feb";"Mar";"Apr";"Maj";"Jun";"Jul";"Aŭg";"Sep";"Okt";"Nov";"Dec"
+abmon	"<U004A><U0061><U006E>";"<U0046><U0065><U0062>";/
+	"<U004D><U0061><U0072>";"<U0041><U0070><U0072>";/
+	"<U004D><U0061><U006A>";"<U004A><U0075><U006E>";/
+	"<U004A><U0075><U006C>";"<U0041><U016D><U0067>";/
+	"<U0053><U0065><U0070>";"<U004F><U006B><U0074>";/
+	"<U004E><U006F><U0076>";"<U0044><U0065><U0063>"
+
+% "Januaro";"Februaro";"Marto";"Aprilo";"Majo";"Junio";
+% "Julio";"Aŭgusto";"Septembro";"Oktobro";"Novembro";"Decembro"
+mon	"<U004A><U0061><U006E><U0075><U0061><U0072><U006F>";/
+	"<U0046><U0065><U0062><U0072><U0075><U0061><U0072><U006F>";/
+	"<U004D><U0061><U0072><U0074><U006F>";/
+	"<U0041><U0070><U0072><U0069><U006C><U006F>";/
+	"<U004D><U0061><U006A><U006F>";/
+	"<U004A><U0075><U006E><U0069><U006F>";/
+	"<U004A><U0075><U006C><U0069><U006F>";/
+	"<U0041><U016D><U0067><U0075><U0073><U0074><U006F>";/
+	"<U0053><U0065><U0070><U0074><U0065><U006D><U0062><U0072><U006F>";/
+	"<U004F><U006B><U0074><U006F><U0062><U0072><U006F>";/
+	"<U004E><U006F><U0076><U0065><U006D><U0062><U0072><U006F>";/
+	"<U0044><U0065><U0063><U0065><U006D><U0062><U0072><U006F>"
+
+am_pm	"";""
+
+% "%a %d %b %Y %T %z"
+d_t_fmt	"<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U007A>"
+
+% "%Y-%m-%d"
+d_fmt	"<U0025><U0059><U002D><U0025><U006D><U002D><U0025><U0064>"
+
+% "%T"
+t_fmt	"<U0025><U0054>"
+
+t_fmt_ampm ""
+
+% date_fmt ?
+
+END LC_TIME
+
+LC_MESSAGES
+% "^[JjYy]"
+yesexpr 	"<U005E><U005B><U004A><U006A><U0059><U0079><U005D>"
+% "^[Nn]"
+noexpr 	"<U005E><U005B><U006E><U004E><U005D>"
+END LC_MESSAGES
+
+LC_PAPER
+height	297
+width	210
+END LC_PAPER
+
+LC_TELEPHONE
+% "+%c %a %l"
+tel_int_fmt	"<U002B><U0025><U0063><U0020><U0025><U0061><U0020>/
+<U0025><U006C>"
+% "00"
+int_select	"<U0030><U0030>"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+measurement    1
+END LC_MEASUREMENT
+
+LC_NAME
+% "%d%t%g%t%m%t%f"
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+name_mr     "<U0073><U002D><U0072><U006F>"              % "s-ro"
+name_mrs    "<U0073><U002D><U0069><U006E><U006F>"       % "s-ino"
+name_ms     "<U0073><U002D><U0069><U006E><U006F>"       % "s-ino"
+END LC_NAME
+
+LC_ADDRESS
+% "%f%N%a%N%d%N%b%N%s %h %e %r%N%%z %T%N%c%N"
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+% "Esperanto"
+lang_name     "<U0045><U0073><U0070><U0065><U0072><U0061><U006E><U0074><U006F>"
+lang_ab       "<U0065><U006F>" % "eo"
+lang_term     "<U0065><U0070><U006F>" % "epo"
+END LC_ADDRESS
diff -urN glibc-2.6.org/localedata/locales/eu_FR glibc-2.6/localedata/locales/eu_FR
--- glibc-2.6.org/localedata/locales/eu_FR	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/eu_FR	2007-05-18 12:14:53.726513832 +0200
@@ -0,0 +1,97 @@
+comment_char %
+escape_char  /
+%
+% Basque Language Locale for France
+% Language: eu
+% Territory: FR
+% Revision: 1.0
+% Date: 2004-06-24
+% Users: general
+% Charset: ISO-8859-1
+% Distribution and use is free, also
+% for commercial purposes.
+
+LC_IDENTIFICATION
+title      "Basque locale for France"
+source     "Christian Perrier and the Debian Project"
+address    ""
+contact    "Christian Perrier"
+email      "bubulle@debian.org"
+tel        ""
+fax        ""
+language   "Basque"
+territory  "France"
+revision   "1.0"
+date       "2004-06-24"
+%
+category  "eu_FR:2000";LC_IDENTIFICATION
+category  "eu_FR:2000";LC_CTYPE
+category  "eu_FR:2000";LC_COLLATE
+category  "eu_FR:2000";LC_TIME
+category  "eu_FR:2000";LC_NUMERIC
+category  "eu_FR:2000";LC_MONETARY
+category  "eu_FR:2000";LC_MESSAGES
+category  "eu_FR:2000";LC_PAPER
+category  "eu_FR:2000";LC_NAME
+category  "eu_FR:2000";LC_ADDRESS
+category  "eu_FR:2000";LC_TELEPHONE
+
+END LC_IDENTIFICATION
+
+LC_COLLATE
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_CTYPE
+copy "i18n"
+END LC_CTYPE
+
+LC_MESSAGES
+copy "eu_ES"
+END LC_MESSAGES
+
+LC_MONETARY
+copy "fr_FR"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "fr_FR"
+END LC_NUMERIC
+
+LC_TIME
+copy "eu_ES"
+END LC_TIME
+
+LC_PAPER
+copy "fr_FR"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "fr_FR"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "fr_FR"
+END LC_MEASUREMENT
+
+LC_NAME
+copy "eu_ES"
+END LC_NAME
+
+LC_ADDRESS
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+country_name "<U0046><U0072><U0061><U006E><U0074><U007A><U0069><U0061>"
+country_ab2  "<U0046><U0052>"
+country_ab3  "<U0046><U0052><U0041>"
+country_num  250
+country_car  "<U0046>"
+country_isbn "<U0032>"
+lang_name    "<U0045><U0075><U0073><U006B><U0061><U0072><U0061>"
+lang_ab      "<U0065><U0075>"
+lang_term    "<U0065><U0075><U0073>"
+lang_lib     "<U0062><U0061><U0071>"
+END LC_ADDRESS
diff -urN glibc-2.6.org/localedata/locales/eu_FR@euro glibc-2.6/localedata/locales/eu_FR@euro
--- glibc-2.6.org/localedata/locales/eu_FR@euro	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/eu_FR@euro	2007-05-18 12:14:53.726513832 +0200
@@ -0,0 +1,84 @@
+comment_char %
+escape_char  /
+%
+% Basque Language Locale for France
+% Language: eu
+% Territory: FR
+% Option: euro
+% Revision: 1.0
+% Date: 2004-06-24
+% Users: general
+% Charset: ISO-8859-15
+% Distribution and use is free, also
+% for commercial purposes.
+
+LC_IDENTIFICATION
+title      "Basque locale for France"
+source     "Christian Perrier and the Debian Project"
+address    ""
+contact    "Christian Perrier"
+email      "bubulle@debian.org"
+tel        ""
+fax        ""
+language   "Basque"
+territory  "France"
+revision   "1.0"
+date       "2004-06-24"
+%
+category  "eu_FR:2000";LC_IDENTIFICATION
+category  "eu_FR:2000";LC_CTYPE
+category  "eu_FR:2000";LC_COLLATE
+category  "eu_FR:2000";LC_TIME
+category  "eu_FR:2000";LC_NUMERIC
+category  "eu_FR:2000";LC_MONETARY
+category  "eu_FR:2000";LC_MESSAGES
+category  "eu_FR:2000";LC_PAPER
+category  "eu_FR:2000";LC_NAME
+category  "eu_FR:2000";LC_ADDRESS
+category  "eu_FR:2000";LC_TELEPHONE
+
+END LC_IDENTIFICATION
+
+LC_COLLATE
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_CTYPE
+copy "i18n"
+END LC_CTYPE
+
+LC_MESSAGES
+copy "eu_FR"
+END LC_MESSAGES
+
+LC_MONETARY
+copy "eu_FR"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "eu_FR"
+END LC_NUMERIC
+
+LC_TIME
+copy "eu_FR"
+END LC_TIME
+
+LC_PAPER
+copy "eu_FR"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "eu_FR"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "eu_FR"
+END LC_MEASUREMENT
+
+LC_NAME
+copy "eu_FR"
+END LC_NAME
+
+LC_ADDRESS
+copy "eu_FR"
+END LC_ADDRESS
diff -urN glibc-2.6.org/localedata/locales/ia glibc-2.6/localedata/locales/ia
--- glibc-2.6.org/localedata/locales/ia	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/ia	2007-05-18 12:14:53.726513832 +0200
@@ -0,0 +1,139 @@
+comment_char %
+escape_char  /
+%
+% Interlingua Language Locale 
+
+LC_IDENTIFICATION
+title      "Interlingua locale"
+source     ""
+address    ""
+contact    ""
+email      "mardy@despammed.com"
+tel        ""
+fax        ""
+language   "Interlingua"
+territory  ""
+revision   "1.0"
+date       "2003-11-25"
+%
+category  "ia:2000";LC_IDENTIFICATION
+category  "ia:2000";LC_CTYPE
+category  "ia:2000";LC_COLLATE
+category  "ia:2000";LC_TIME
+category  "ia:2000";LC_NUMERIC
+category  "ia:2000";LC_MONETARY
+category  "ia:2000";LC_MESSAGES
+category  "ia:2000";LC_PAPER
+category  "ia:2000";LC_NAME
+category  "ia:2000";LC_ADDRESS
+category  "ia:2000";LC_TELEPHONE
+
+END LC_IDENTIFICATION
+
+LC_COLLATE
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_CTYPE
+copy "i18n"
+END LC_CTYPE
+
+LC_MESSAGES
+yesexpr "<U005E><U005B><U0073><U0053><U0079><U0059><U005D><U002E><U002A>"
+noexpr  "<U005E><U005B><U006E><U004E><U005D><U002E><U002A>"
+END LC_MESSAGES
+
+LC_MONETARY
+int_curr_symbol           "<U0045><U0055><U0052><U0020>"
+currency_symbol           "<U20AC>"
+mon_decimal_point         "<U002C>"
+mon_thousands_sep         "<U002E>"
+mon_grouping              3;3
+positive_sign             ""
+negative_sign             "<U002D>"
+int_frac_digits           2
+frac_digits               2
+p_cs_precedes             1
+p_sep_by_space            1
+n_cs_precedes             1
+n_sep_by_space            1
+p_sign_posn               1
+n_sign_posn               1
+END LC_MONETARY
+
+LC_NUMERIC
+decimal_point             "<U002C>"
+thousands_sep             ""
+grouping                  0;0
+END LC_NUMERIC
+
+LC_TIME
+abday   "<U0064><U006F><U006D>";"<U006C><U0075><U006E>";/
+        "<U006D><U0061><U0072>";"<U006D><U0065><U0072>";/
+        "<U006A><U006F><U0076>";"<U0076><U0065><U006E>";/
+        "<U0073><U0061><U0062>"
+day     "<U0064><U006F><U006D><U0069><U006E><U0069><U0063><U0061>";/
+        "<U006C><U0075><U006E><U0065><U0064><U0069>";/
+        "<U006D><U0061><U0072><U0074><U0065><U0064><U0069>";/
+        "<U006D><U0065><U0072><U0063><U0075><U0072><U0069><U0064><U0069>";/
+        "<U006A><U006F><U0076><U0065><U0064><U0069>";/
+        "<U0076><U0065><U006E><U0065><U0072><U0064><U0069>";/
+        "<U0073><U0061><U0062><U0062><U0061><U0074><U006F>"
+abmon   "<U006A><U0061><U006E>";"<U0066><U0065><U0062>";/
+        "<U006D><U0061><U0072>";"<U0061><U0070><U0072>";/
+        "<U006D><U0061><U0069>";"<U006A><U0075><U006E>";/
+        "<U006A><U0075><U006C>";"<U0061><U0075><U0067>";/
+        "<U0073><U0065><U0070>";"<U006F><U0063><U0074>";/
+        "<U006E><U006F><U0076>";"<U0064><U0069><U0063>"
+mon     "<U006A><U0061><U006e><U0075><U0061><U0072><U0069><U006f>";/
+        "<U0066><U0065><U0062><U0072><U0075><U0061><U0072><U0069><U006F>";/
+        "<U006D><U0061><U0072><U0074><U0069><U006F>";/
+        "<U0061><U0070><U0072><U0069><U006C>";/
+        "<U006D><U0061><U0069><U006F>";/
+        "<U006A><U0075><U006E><U0069><U006F>";/
+        "<U006A><U0075><U006C><U0069><U006F>";/
+        "<U0061><U0075><U0067><U0075><U0073><U0074><U006F>";/
+        "<U0073><U0065><U0070><U0074><U0065><U006D><U0062><U0072><U0065>";/
+        "<U006f><U0063><U0074><U006F><U0062><U0072><U0065>";/
+        "<U006E><U006F><U0076><U0065><U006D><U0062><U0072><U0065>";/
+        "<U0064><U0065><U0063><U0065><U006D><U0062><U0072><U0065>"
+d_t_fmt "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+d_fmt   "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
+t_fmt   "<U0025><U0054>"
+am_pm   "";""
+t_fmt_ampm ""
+date_fmt       "<U0025><U0061><U0020><U0025><U0062><U0020><U0025><U0065>/
+<U0020><U0025><U0048><U003A><U0025><U004D><U003A><U0025><U0053><U0020>/
+<U0025><U005A><U0020><U0025><U0059>"
+END LC_TIME
+
+LC_PAPER
+% FIXME
+height   297
+% FIXME
+width    210
+END LC_PAPER
+
+LC_TELEPHONE
+tel_int_fmt    "<U002B><U0025><U0063><U0020><U0025><U0061><U0020><U0025>/
+<U006C>"
+int_select	"<U0030><U0030>"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+% FIXME
+measurement    1
+END LC_MEASUREMENT
+
+LC_NAME
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+END LC_NAME
+
+LC_ADDRESS
+postal_fmt    "<U0025><U0066><U0025><U004E><U0025><U0061><U0025><U004E>/
+<U0025><U0064><U0025><U004E><U0025><U0062><U0025><U004E><U0025><U0073>/
+<U0020><U0025><U0068><U0020><U0025><U0065><U0020><U0025><U0072><U0025>/
+<U004E><U0025><U0025><U007A><U0020><U0025><U0054><U0025>/
+<U004E><U0025><U0063><U0025><U004E>"
+END LC_ADDRESS
diff -urN glibc-2.6.org/localedata/locales/nds_DE@traditional glibc-2.6/localedata/locales/nds_DE@traditional
--- glibc-2.6.org/localedata/locales/nds_DE@traditional	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/nds_DE@traditional	2007-05-18 12:14:53.729847465 +0200
@@ -0,0 +1,116 @@
+comment_char %
+escape_char  /
+repertoiremap mnemonic.ds
+% charset "ISO_8859-1:1987"
+% Distribution and use is
+
+LC_IDENTIFICATION 
+title     "Low(lands) Saxon Language Locale for Germany"
+source    "information from Kenneth Christiansen"
+address   ""
+contact   "Kenneth Christiansen, Pablo Saratxaga"
+email     "kenneth@gnu.org, pablo@mandrakesoft.com"
+tel       ""
+fax       ""
+language  "Low Saxon"
+territory "Germany"
+revision  "0.1"
+date      "2003-11-30"
+
+category i18n:2000;LC_IDENTIFICATION
+category i18n:2000;LC_COLLATE
+category i18n:2000;LC_CTYPE
+category i18n:2000;LC_MESSAGES
+category i18n:2000;LC_MONETARY
+category i18n:2000;LC_NUMERIC
+category i18n:2000;LC_TIME
+category i18n:2000;LC_PAPER
+category i18n:2000;LC_NAME
+category i18n:2000;LC_MEASUREMENT
+category i18n:2000;LC_TELEPHONE
+category i18n:2000;LC_ADDRESS
+
+% ISO/IEC 14652: Converted comments to LC_IDENTIFICATION
+%  added LC_PAPER, LC_MEASUREMENT, LC_TELEPHONE, LC_ADDRESS
+%  copied LC_MONETARY, LC_NUMERIC from de_DE for simplicity
+% -- mckinstry@computer.org, 2000-07-03
+
+END LC_IDENTIFICATION
+
+LC_ADDRESS
+copy "nds_DE"
+END LC_ADDRESS
+
+LC_COLLATE
+copy "nds_DE"
+END LC_COLLATE
+
+LC_CTYPE
+copy "nds_DE"
+END LC_CTYPE
+
+LC_MESSAGES
+copy "nds_DE"
+END LC_MESSAGES
+
+LC_MONETARY
+copy "nds_DE"
+END LC_MONETARY
+
+LC_PAPER
+copy "nds_DE"
+END LC_PAPER
+
+LC_NAME
+name_fmt    "<U0025><U0064><U0025><U0074><U0025><U0067><U0025><U0074>/
+<U0025><U006D><U0025><U0074><U0025><U0066>"
+END LC_NAME
+
+LC_TELEPHONE
+copy "nds_DE"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "nds_DE"
+END LC_MEASUREMENT
+
+LC_NUMERIC
+copy "nds_DE"
+END LC_NUMERIC
+
+LC_TIME
+abday   "<U0053><U0064><U0061><U0067>";"<U004D><U0061><U0061><U006E>";/
+        "<U0044><U0069><U006E><U0067>";"<U004D><U0069><U0067><U0067>";/
+        "<U0044><U0075><U006E><U006E>";"<U0046><U0072><U0065><U0065>";/
+        "<U0053><U0076><U0064><U002E>"
+day     "<U0053><U00FC><U006E><U006E><U0064><U0061><U0067>";/
+        "<U004D><U0061><U0061><U006E><U0064><U0061><U0067>";/
+        "<U0044><U0069><U006E><U0067><U0073><U0064><U0061><U0067>";/
+        "<U004D><U0069><U0064><U0064><U0065><U0077><U0065><U0065><U006B>";/
+        "<U0044><U0075><U006E><U006E><U0065><U0072><U0073><U0064><U0061><U0067>";/
+        "<U0046><U0072><U0065><U0065><U0064><U0061><U0067>";/
+        "<U0053><U00FC><U006E><U006E><U0061><U0076><U0065><U006E><U0064>"
+abmon   "<U0053><U006E><U0065><U0065>";"<U0049><U0065><U0073><U002E>";/
+        "<U004C><U0065><U006E><U0074>";"<U004F><U006F><U0073><U0074>";/
+        "<U004D><U0061><U0069><U002E>";"<U0042><U0072><U0061><U0061>";/
+        "<U0048><U0061><U0075><U002E>";"<U0041><U0075><U0067><U002E>";/
+        "<U0048><U0061><U0072><U0066>";"<U0057><U0069><U0065><U006E>";/
+        "<U004E><U0065><U0076><U002E>";"<U0043><U0068><U0072><U0069>"
+mon     "<U0053><U006E><U0065><U0065><U006D><U0061><U0061><U006E><U0064>";/
+        "<U0049><U0065><U0073><U006D><U0061><U0061><U006E><U0064>";/
+        "<U004C><U0065><U006E><U0074><U006D><U0061><U0061><U006E><U0064>";/
+        "<U004F><U006F><U0073><U0074><U0065><U0072><U006D><U0061><U0061><U006E><U0064>";/
+        "<U004D><U0061><U0069><U006D><U0061><U0061><U006E><U0064>";/
+        "<U0042><U0072><U0061><U0061><U006B><U006D><U0061><U0061><U006E><U0064>";/
+        "<U0048><U0061><U0075><U006D><U0061><U0061><U006E><U0064>";/
+        "<U0041><U0075><U0067><U0073><U0074><U006D><U0061><U0061><U006E><U0064>";/
+        "<U0048><U0061><U0072><U0066><U0073><U0074><U006D><U0061><U0061><U006E><U0064>";/
+        "<U0057><U0069><U0065><U006E><U006D><U0061><U0061><U006E><U0064>";/
+        "<U004E><U0065><U0076><U0065><U006C><U006D><U0061><U0061><U006E><U0064>";/
+        "<U0043><U0068><U0072><U0069><U0073><U0074><U006D><U0061><U0061><U006E><U0064>"
+d_t_fmt "<U0025><U0061><U0020><U0025><U0064><U002E><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
+d_fmt   "<U0025><U0064><U002E><U0025><U006D><U002E><U0025><U0059>"
+t_fmt   "<U0025><U0054>"
+am_pm   "";""
+t_fmt_ampm ""
+END LC_TIME
diff -urN glibc-2.6.org/localedata/locales/oc_FR@euro glibc-2.6/localedata/locales/oc_FR@euro
--- glibc-2.6.org/localedata/locales/oc_FR@euro	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/oc_FR@euro	2007-05-18 12:14:53.733181099 +0200
@@ -0,0 +1,83 @@
+comment_char %
+escape_char  /
+%
+% Occitan Language Locale for France with Euro
+% Language: oc
+% Territory: FR
+% Option: euro
+% Date: 2004-11-23
+% Users: general
+% Charset: ISO-8859-15
+% Distribution and use is free, also
+% for commercial purposes.
+
+LC_IDENTIFICATION
+title      "Occitan Language Locale for France with Euro"
+source     "Free Software Foundation, Inc."
+address    "59 Temple Place - Suite 330, Boston, MA 02111-1307, USA"
+contact    ""
+email      "bug-glibc@gnu.org"
+tel        ""
+fax        ""
+language   "Occitan"
+territory  "France"
+revision   "1.0"
+date       "2004-11-23"
+%
+category  "i18n:2000";LC_IDENTIFICATION
+category  "i18n:2000";LC_CTYPE
+category  "i18n:2000";LC_COLLATE
+category  "i18n:2000";LC_TIME
+category  "i18n:2000";LC_NUMERIC
+category  "i18n:2000";LC_MONETARY
+category  "i18n:2000";LC_MESSAGES
+category  "i18n:2000";LC_PAPER
+category  "i18n:2000";LC_NAME
+category  "i18n:2000";LC_ADDRESS
+category  "i18n:2000";LC_TELEPHONE
+
+END LC_IDENTIFICATION
+
+LC_COLLATE
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_MESSAGES
+copy "oc_FR"
+END LC_MESSAGES
+
+LC_MONETARY
+copy "oc_FR"
+END LC_MONETARY
+
+LC_CTYPE
+copy "oc_FR"
+END LC_CTYPE
+
+LC_NUMERIC
+copy "oc_FR"
+END LC_NUMERIC
+
+LC_TIME
+copy "oc_FR"
+END LC_TIME
+
+LC_PAPER
+copy "oc_FR"
+END LC_PAPER
+
+LC_TELEPHONE
+copy "oc_FR"
+END LC_TELEPHONE
+
+LC_MEASUREMENT
+copy "oc_FR"
+END LC_MEASUREMENT
+
+LC_NAME
+copy "oc_FR"
+END LC_NAME
+
+LC_ADDRESS
+copy "oc_FR"
+END LC_ADDRESS
diff -urN glibc-2.6.org/localedata/locales/rm_CH glibc-2.6/localedata/locales/rm_CH
--- glibc-2.6.org/localedata/locales/rm_CH	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/rm_CH	2007-05-18 12:14:53.733181099 +0200
@@ -0,0 +1,193 @@
+comment_char %
+escape_char  /
+% Locale for Rhaeto-Romance language in Switzerland (Romansh).
+% Contributed by Sascha Brawer <brawer at dandelis.ch>, 2003
+% corrected by Florian Verdet and Reto Lehner, 2003
+% internal-revision: 0.1.2  2003-11-18
+
+LC_IDENTIFICATION
+%fvg:I assume, this is correct...
+title      "Rhaeto-Romance locale for Switzerland"
+source     "Free Software Foundation, Inc."
+address    "59 Temple Place - Suite 330, Boston, MA 02111-1307, USA"
+contact    "Sascha Brawer"
+email      "brawer at dandelis.ch"
+tel        ""
+fax        ""
+language   "Rhaeto-Romance"
+territory  "Switzerland"
+revision   "0.1"
+date       "2003-11-12"
+%
+category  "rm_CH:2000";LC_IDENTIFICATION
+category  "rm_CH:2000";LC_CTYPE
+category  "rm_CH:2000";LC_COLLATE
+category  "rm_CH:2000";LC_TIME
+category  "rm_CH:2000";LC_NUMERIC
+category  "rm_CH:2000";LC_MONETARY
+category  "rm_CH:2000";LC_MESSAGES
+category  "rm_CH:2000";LC_PAPER
+category  "rm_CH:2000";LC_NAME
+category  "rm_CH:2000";LC_ADDRESS
+category  "rm_CH:2000";LC_TELEPHONE
+
+END LC_IDENTIFICATION
+
+LC_COLLATE
+copy "iso14651_t1"
+END LC_COLLATE
+
+LC_CTYPE
+% FIXME: This assumes that Romansh uses the same transliterations
+% as German. Is this correct?
+%fvg:FALSE!
+%-uuml, ouml (and auml) only occur in names
+% they occur also in idioms, but not in RG !
+%-[aeiou]grave are used !
+%-'s-ch' (one collating-symbol ?)(occurs also in RG ??) is collated with 'sch'
+% (scha, schabain, s-chabè, schablà, schablona, s-chablun, schacal, ...).)
+copy "de_CH"
+END LC_CTYPE
+
+LC_MESSAGES
+%fvg: idioms use: eEhHsS (ea,schi,hai) - and nNbB (na,bu?,betg?)
+yesexpr                   "^[gGyY]"  % User input for accepting.
+noexpr                    "^[nN]"    % User input for rejecting.
+END LC_MESSAGES
+
+LC_MONETARY
+int_curr_symbol           "CHF "
+currency_symbol           "Fr."
+mon_decimal_point         "<U002E>"	%FULL STOP
+mon_thousands_sep         "<U0020>"	%SPACE - fvg:Why not an apastroph ?
+mon_grouping              3;3
+positive_sign             ""
+negative_sign             "<U002D>"	%HYPHEN-MINUS
+int_frac_digits           2
+frac_digits               2
+p_cs_precedes             1
+p_sep_by_space            1
+n_cs_precedes             1
+n_sep_by_space            1
+p_sign_posn               4
+n_sign_posn               4
+END LC_MONETARY
+
+LC_NUMERIC
+decimal_point             "<U002E>"
+thousands_sep             "<U0020>"	%fvg:dito
+grouping                  3;3
+END LC_NUMERIC
+
+LC_TIME
+%fvg:dictionary says: mixed two-char/three-char abbrev:
+%abday   "du"; "gli"; "ma"; "me"; "gie"; "ve"; "so"
+%but specification wants "of equal length" !
+abday   "dum"; "gli"; "mar"; "mes"; "gie"; "ven"; "son"
+day     "dumengia"; "glindesdi"; "mardi"; /
+        "mesemna"; "gievgia"; "venderdi"; "sonda"
+%7 days in a week ; date of a Monday (firt day of week) ;
+%week of first Thursday of a year is week #1  } is ISO 8601 conform
+week    7;19971201;4
+%fvg:abbrev not confirmed
+abmon   "sch"; "fav"; "mar"; "avr"; "mat"; "zer";/
+        "fan"; "avu"; "set"; "oct"; "nov"; "dec"
+mon     "schaner"; "favrer"; "mars";/
+        "avrigl"; "matg"; "zercladur";/
+        "fanadur"; "avust"; "settember";/
+        "october"; "november"; "december"
+am_pm   "";""                % AM and PM are not appended.
+d_t_fmt "%a %d %b %Y %T %Z"  % Date and time respresentation (%c).
+d_fmt   "%d-%m-%Y"           % Date representation (%x).
+t_fmt   "%T"                 % Time representation (%X).
+t_fmt_ampm ""                % No AM/PM suffix.
+date_fmt "%a %e %b %H:%M:%S" % Full date.
+END LC_TIME
+
+LC_PAPER
+height   297  % Height of normal sheet in mm (DIN A4).
+width    210  % Width of normal sheet in mm (DIN A4).
+END LC_PAPER
+
+LC_TELEPHONE
+tel_int_fmt    "+%c %a %l"
+int_prefix     "41"
+END LC_TELEPHONE
+
+
+LC_MEASUREMENT
+measurement    1  % metric system
+END LC_MEASUREMENT
+
+LC_NAME
+
+% Format of a name, taken from the glibc locale for Ukrainian.
+% %f Family names.
+% %F Family names in uppercase. 
+% %g First given name.
+% %G First given initial.
+% %l First given name with Latin letters.
+% %o Other shorter name.
+% %m Additional given names.
+% %M Initials for additional given names.
+% %p Profession.
+% %s Salutation, such as "Doctor"
+% %S Abbreviated salutation, such as "Mr." or "Dr."
+% %d Salutation, using the FDCC-sets conventions
+%    1 for the name_gen
+%    2 for name_mr
+%    3 for name_mrs
+%    4 for name_miss
+%    5 for name_ms
+% %t If the preceding field descriptor resulted in an empty string,
+%    then the empty string, else a <space>.
+
+name_fmt    "%d%t%g%t%m%t%f"
+name_gen    ""         % A salutation valid for all persons.
+name_miss   "Dunna"    % A salutation valid for unmarried females.
+%'Signur' is not really used any more
+name_mr     "Sar"      % A salutation valid for males.
+name_mrs    "Dunna"    % A salutation valid for married females.
+name_ms     "Dunna"    % A salutation valid for all females.
+%all these should be written lower-case if they occur inside a phrase !
+END LC_NAME
+
+LC_ADDRESS
+
+% Format of an address. The fields have the following meaning
+% (taken from glibc locale for Ukranian):
+
+% %n Person's name, possibly constructed with the LC_NAME
+%    "name_fmt" keyword. 
+% %a Care of person, or organization. 
+% %f Firm name. 
+% %d Department name. 
+% %b Building name. 
+% %s Street or block (eg. Japanese) name. 
+% %h House number or designation. 
+% %N Insert an <end-of-line> if the previous descriptor
+%    value was not an empty string; otherwise ignore.
+% %t Insert a <space> if the previous descriptor value
+%    was not an empty string; otherwise ignore.
+% %r Room number, door designation. 
+% %e Floor number. 
+% %C Country designation, from the <country_post> keyword. 
+% %l Local township within town or city .
+% %z Zip number, postal code. 
+% %T Town, city. 
+% %S State, province, or prefecture. 
+% %c Country, as taken from data record.
+
+postal_fmt   "%f%N%a%N%d%N%b%N%s %h%t%e%t%r%N%z %T%N%c%N"
+country_name "Svizra"          % Localized country name.
+country_post "<U0043><U0048>"  % Postal code for country.
+country_num  756               % Numeric code for country (ISO 3166).
+country_ab2  "<U0043><U0048>"  % Two-letter code for country (ISO 3166).
+country_ab3  "<U0043><U0048><U0045>" % Three-letter code for country (ISO 3166).
+country_car  "<U0043><U0048>"  % Country code for cars.
+country_isbn "<U0038><U0038>"  % same as for Italian ?
+lang_name    "rumantsch"       % Localized language name.
+lang_ab      "<U0072><U006D>"  % Two-letter code for language (ISO 639).
+lang_term    "<U0072><U006F><U0068>" % Three-letter code for language (ISO 639-2).
+lang_lib     "<U0072><U006F><U0068>" % Three-letter code for language (ISO 639-2).
+END LC_ADDRESS
diff -urN glibc-2.6.org/localedata/locales/sw_KE glibc-2.6/localedata/locales/sw_KE
--- glibc-2.6.org/localedata/locales/sw_KE	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/sw_KE	2007-05-18 12:14:58.623621321 +0200
@@ -0,0 +1,355 @@
+comment_char %
+escape_char  /
+%
+% Swahili Language Locale for Kenya
+% Source: Based on data from Debian/Mandriva
+% Contact: ankry at mif dot pg dot gda dot pl
+% Email: 
+% Language: sw
+% Territory: Kenya
+% Revision: 0.3
+% Date: 2005-10-14
+% Users: general
+% Repertoiremap: mnemonic,ds
+% Charset: UTF-8
+% Distribution and use is free, also
+% for commercial purposes.
+
+LC_IDENTIFICATION
+title      "Swahili locale for Kenya"
+source     "Based on data from Debian/Mandriva"
+address    ""
+contact    ""
+email      ""
+tel        ""
+fax        ""
+language   "Swahili"
+territory  "Kenya"
+revision   "0.2"
+date       "2004-08-01"
+%
+category  "i18n:1997";LC_IDENTIFICATION
+category  "i18n:1997";LC_CTYPE
+category  "i18n:1997";LC_COLLATE
+category  "i18n:1997";LC_TIME
+category  "i18n:1997";LC_NUMERIC
+category  "i18n:1997";LC_MONETARY
+category  "i18n:1997";LC_MESSAGES
+category  "i18n:1997";LC_PAPER
+category  "i18n:1997";LC_NAME
+category  "i18n:1997";LC_ADDRESS
+category  "i18n:1997";LC_TELEPHONE
+END LC_IDENTIFICATION
+
+%%%%%%%%%%%%%%%%
+% TODO: check LC_MONETARY, LC_NUMERIC, LC_MESSAGES, LC_COLLATE
+% 	check time and date formats
+%%%%%%%%%%%%%%%%
+
+LC_COLLATE
+% Copy the template from ISO/IEC 14651 i.e.
+% use the rules there when making ordered lists of words.
+%
+copy "iso14651_t1"
+% a b c ch d dh e f g gh h i j k kh l m mb mv n nd ng ng' nj ny nz o p q r
+% s sh t th u v w x y z
+
+collating-symbol  <ch>
+collating-element <C-H> from "<U0043><U0048>"
+collating-element <C-h> from "<U0043><U0068>"
+collating-element <c-H> from "<U0063><U0048>"
+collating-element <c-h> from "<U0063><U0068>"
+
+collating-symbol  <dh>
+collating-element <D-H> from "<U0044><U0048>"
+collating-element <D-h> from "<U0044><U0068>"
+collating-element <d-H> from "<U0064><U0048>"
+collating-element <d-h> from "<U0064><U0068>"
+
+collating-symbol  <gh>
+collating-element <G-H> from "<U0047><U0043>"
+collating-element <G-h> from "<U0047><U0063>"
+collating-element <g-H> from "<U0067><U0043>"
+collating-element <g-h> from "<U0067><U0063>"
+
+collating-symbol  <kh>
+collating-element <K-H> from "<U004B><U0048>"
+collating-element <K-h> from "<U004B><U0068>"
+collating-element <k-H> from "<U006B><U0048>"
+collating-element <k-h> from "<U006B><U0068>"
+
+collating-symbol  <mb>
+collating-element <M-B> from "<U004D><U0042>"
+collating-element <M-b> from "<U004D><U0062>"
+collating-element <m-B> from "<U006D><U0042>"
+collating-element <m-b> from "<U006D><U0062>"
+
+collating-symbol  <mv>
+collating-element <M-V> from "<U004D><U0056>"
+collating-element <M-v> from "<U004D><U0076>"
+collating-element <m-V> from "<U006D><U0056>"
+collating-element <m-v> from "<U006D><U0076>"
+
+collating-symbol  <nd>
+collating-element <N-D> from "<U004E><U0044>"
+collating-element <N-d> from "<U004E><U0064>"
+collating-element <n-D> from "<U006E><U0044>"
+collating-element <n-d> from "<U006E><U0064>"
+
+collating-symbol  <ng>
+collating-element <N-G> from "<U004E><U0047>"
+collating-element <N-g> from "<U004E><U0067>"
+collating-element <n-G> from "<U006E><U0047>"
+collating-element <n-g> from "<U006E><U0067>"
+
+collating-symbol  <ng'>
+collating-element <N-G-'> from "<U004E><U0047><U0027>"
+collating-element <N-g-'> from "<U004E><U0067><U0027>"
+collating-element <n-G-'> from "<U006E><U0047><U0027>"
+collating-element <n-g-'> from "<U006E><U0067><U0027>"
+
+collating-symbol  <nj>
+collating-element <N-J> from "<U004E><U004A>"
+collating-element <N-j> from "<U004E><U006A>"
+collating-element <n-J> from "<U006E><U004A>"
+collating-element <n-j> from "<U006E><U006A>"
+
+collating-symbol  <ny>
+collating-element <N-Y> from "<U004E><U0059>"
+collating-element <N-y> from "<U004E><U0079>"
+collating-element <n-Y> from "<U006E><U0059>"
+collating-element <n-y> from "<U006E><U0079>"
+
+collating-symbol  <nz>
+collating-element <N-Z> from "<U004E><U005A>"
+collating-element <N-z> from "<U004E><U007A>"
+collating-element <n-Z> from "<U006E><U005A>"
+collating-element <n-z> from "<U006E><U007A>"
+
+collating-symbol  <sh>
+collating-element <S-H> from "<U0053><U0048>"
+collating-element <S-h> from "<U0053><U0068>"
+collating-element <s-H> from "<U0073><U0048>"
+collating-element <s-h> from "<U0073><U0068>"
+
+% symbol <th> is already used in iso14651_t1
+collating-symbol  <th0>
+collating-element <T-H> from "<U0054><U0048>"
+collating-element <T-h> from "<U0054><U0068>"
+collating-element <t-H> from "<U0074><U0048>"
+collating-element <t-h> from "<U0074><U0068>"
+
+collating-symbol <CAP-CAP>
+collating-symbol <CAP-MIN>
+collating-symbol <MIN-CAP>
+collating-symbol <MIN-MIN>
+
+reorder-after <MIN>
+<MIN-MIN>
+<MIN-CAP>
+
+reorder-after <CAP>
+<CAP-MIN>
+<CAP-CAP>
+
+reorder-after <c>
+<ch>
+reorder-after <d>
+<dh>
+reorder-after <g>
+<gh>
+reorder-after <k>
+<kh>
+reorder-after <m>
+<mb>
+<mv>
+reorder-after <n>
+<nd>
+<ng>
+<ng'>
+<nj>
+<ny>
+<nz>
+reorder-after <s>
+<sh>
+reorder-after <t>
+<th0>
+
+reorder-after <U0043>
+<C-h>		<ch>;<BAS>;<CAP-MIN>;IGNORE
+<C-H>		<ch>;<BAS>;<CAP-CAP>;IGNORE
+reorder-after <U0063>
+<c-h>		<ch>;<BAS>;<MIN-MIN>;IGNORE
+<c-H>		<ch>;<BAS>;<MIN-CAP>;IGNORE
+
+reorder-after <U0044>
+<D-h>		<dh>;<BAS>;<CAP-MIN>;IGNORE
+<D-H>		<dh>;<BAS>;<CAP-CAP>;IGNORE
+reorder-after <U0064>
+<d-h>		<dh>;<BAS>;<MIN-MIN>;IGNORE
+<d-H>		<dh>;<BAS>;<MIN-CAP>;IGNORE
+
+reorder-after <U0047>
+<G-h>		<gh>;<BAS>;<CAP-MIN>;IGNORE
+<G-H>		<gh>;<BAS>;<CAP-CAP>;IGNORE
+reorder-after <U0067>
+<g-h>		<gh>;<BAS>;<MIN-MIN>;IGNORE
+<g-H>		<gh>;<BAS>;<MIN-CAP>;IGNORE
+
+reorder-after <U004B>
+<K-h>		<kh>;<BAS>;<CAP-MIN>;IGNORE
+<K-H>		<kh>;<BAS>;<CAP-CAP>;IGNORE
+reorder-after <U006B>
+<k-h>		<kh>;<BAS>;<MIN-MIN>;IGNORE
+<k-H>		<kh>;<BAS>;<MIN-CAP>;IGNORE
+
+reorder-after <U004D>
+<M-b>		<mb>;<BAS>;<CAP-MIN>;IGNORE
+<M-B>		<mb>;<BAS>;<CAP-CAP>;IGNORE
+<M-v>		<mv>;<BAS>;<CAP-MIN>;IGNORE
+<M-V>		<mv>;<BAS>;<CAP-CAP>;IGNORE
+reorder-after <U006D>
+<m-b>		<mb>;<BAS>;<MIN-MIN>;IGNORE
+<m-B>		<mb>;<BAS>;<MIN-CAP>;IGNORE
+<m-v>		<mv>;<BAS>;<MIN-MIN>;IGNORE
+<m-V>		<mv>;<BAS>;<MIN-CAP>;IGNORE
+
+reorder-after <U004E>
+<N-d>		<nd>;<BAS>;<CAP-MIN>;IGNORE
+<N-D>		<nd>;<BAS>;<CAP-CAP>;IGNORE
+<N-g>		<ng>;<BAS>;<CAP-MIN>;IGNORE
+<N-G>		<ng>;<BAS>;<CAP-CAP>;IGNORE
+<N-g-'>		<ng'>;<BAS>;<CAP-MIN>;IGNORE
+<N-G-'>		<ng'>;<BAS>;<CAP-CAP>;IGNORE
+<N-j>		<nj>;<BAS>;<CAP-MIN>;IGNORE
+<N-J>		<nj>;<BAS>;<CAP-CAP>;IGNORE
+<N-y>		<ny>;<BAS>;<CAP-MIN>;IGNORE
+<N-Y>		<ny>;<BAS>;<CAP-CAP>;IGNORE
+<N-z>		<nz>;<BAS>;<CAP-MIN>;IGNORE
+<N-Z>		<nz>;<BAS>;<CAP-CAP>;IGNORE
+reorder-after <U006E>
+<n-d>		<nd>;<BAS>;<MIN-MIN>;IGNORE
+<n-D>		<nd>;<BAS>;<MIN-CAP>;IGNORE
+<n-g>		<ng>;<BAS>;<MIN-MIN>;IGNORE
+<n-G>		<ng>;<BAS>;<MIN-CAP>;IGNORE
+<n-g-'>		<ng'>;<BAS>;<MIN-MIN>;IGNORE
+<n-G-'>		<ng'>;<BAS>;<MIN-CAP>;IGNORE
+<n-j>		<nj>;<BAS>;<MIN-MIN>;IGNORE
+<n-J>		<nj>;<BAS>;<MIN-CAP>;IGNORE
+<n-y>		<ny>;<BAS>;<MIN-MIN>;IGNORE
+<n-Y>		<ny>;<BAS>;<MIN-CAP>;IGNORE
+<n-z>		<nz>;<BAS>;<MIN-MIN>;IGNORE
+<n-Z>		<nz>;<BAS>;<MIN-CAP>;IGNORE
+
+reorder-after <U0053>
+<S-h>		<sh>;<BAS>;<CAP-MIN>;IGNORE
+<S-H>		<sh>;<BAS>;<CAP-CAP>;IGNORE
+reorder-after <U0073>
+<s-h>		<sh>;<BAS>;<MIN-MIN>;IGNORE
+<s-H>		<sh>;<BAS>;<MIN-CAP>;IGNORE
+
+reorder-after <U0054>
+<T-h>		<th0>;<BAS>;<CAP-MIN>;IGNORE
+<T-H>		<th0>;<BAS>;<CAP-CAP>;IGNORE
+reorder-after <U0074>
+<t-h>		<th0>;<BAS>;<MIN-MIN>;IGNORE
+<t-H>		<th0>;<BAS>;<MIN-CAP>;IGNORE
+
+reorder-end
+
+END LC_COLLATE
+
+LC_CTYPE
+copy "i18n"
+END LC_CTYPE
+
+LC_MONETARY
+copy "om_KE"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "om_KE"
+END LC_NUMERIC
+
+LC_TELEPHONE
+copy "om_KE"
+END LC_TELEPHONE
+
+LC_TIME
+abday   "<U004A><U0070><U0069>";"<U004A><U0074><U0061>";/
+        "<U004A><U006E><U0065>";"<U004A><U0074><U006E>";/
+        "<U0041><U006C><U0068>";"<U0049><U006A><U0075>";/
+        "<U004A><U006D><U006F>"
+day     "<U004A><U0075><U006D><U0061><U0070><U0069><U006C><U0069>";/
+        "<U004A><U0075><U006D><U0061><U0074><U0061><U0074><U0075>";/
+        "<U004A><U0075><U006D><U0061><U006E><U006E><U0065>";/
+        "<U004A><U0075><U006D><U0061><U0074><U0061><U006E><U006F>";/
+        "<U0041><U006C><U0068><U0061><U006D><U0069><U0073><U0069>";/
+        "<U0049><U006A><U0075><U006D><U0061><U0061>";/
+        "<U004A><U0075><U006D><U0061><U006D><U006F><U0073><U0069>"
+abmon   "<U004A><U0061><U006E>";"<U0046><U0065><U0062>";/
+        "<U004D><U0061><U0063>";"<U0041><U0070><U0072>";/
+        "<U004D><U0065><U0069>";"<U004A><U0075><U006E>";/
+        "<U004A><U0075><U006C>";"<U0041><U0075><U0067>";/
+        "<U0053><U0065><U0070>";"<U004F><U006B><U0074>";/
+        "<U004E><U006F><U0076>";"<U0044><U0069><U0073>"
+mon     "<U004A><U0061><U006E><U0075><U0061><U0072><U0069>";/
+        "<U0046><U0065><U0062><U0072><U0075><U0061><U0072><U0069>";/
+        "<U004D><U0061><U0063><U0068><U0069>";/
+        "<U0041><U0070><U0072><U0069><U006C><U0069>";/
+        "<U004D><U0065><U0069>";/
+        "<U004A><U0075><U006E><U0069>";/
+        "<U004A><U0075><U006C><U0061><U0069>";/
+        "<U0041><U0075><U0067><U006F><U0073><U0074><U0069>";/
+        "<U0053><U0065><U0070><U0074><U0065><U006D><U0062><U0061>";/
+        "<U004F><U006B><U0074><U006F><U0062><U0061>";/
+        "<U004E><U006F><U0076><U0065><U006D><U0062><U0061>";/
+        "<U0044><U0069><U0073><U0065><U006D><U0062><U0061>"
+%
+d_t_fmt     "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0072><U0020><U0025><U005A>"
+d_fmt       "<U0025><U006D><U002F><U0025><U0064><U002F><U0025><U0079>"
+t_fmt       "<U0025><U0072>"
+am_pm       "<U0041><U004D>";"<U0050><U004D>"
+t_fmt_ampm  "<U0025><U0049><U003A><U0025><U004D><U003A><U0025><U0053><U0020><U0025><U0070>"
+END LC_TIME
+
+LC_MESSAGES
+% FIXME
+copy "en_US"
+END LC_MESSAGES
+
+% Default paper: A4
+LC_PAPER
+height 297
+width 210
+END LC_PAPER
+
+% Metric System (ISO 1000)
+LC_MEASUREMENT
+measurement 1
+END LC_MEASUREMENT
+
+LC_ADDRESS
+%
+% This is the ISO_IEC TR14652 Locale definition for the LC_ADDRESS category.
+%
+postal_fmt    "<U0025><U007A><U0025><U0063><U0025><U0054><U0025><U0073>/
+<U0025><U0062><U0025><U0065><U0025><U0072>"
+%
+country_name  "<U004B><U0065><U006E><U0079><U0061>"
+country_post  "<U004B><U0045><U004E>"
+country_ab2   "<U004B><U0045>"
+country_ab3   "<U004B><U0045><U004E>"
+country_num   404
+country_car   "<U0045><U0041><U004B>"
+country_isbn  "<U0039><U0039><U0036><U0036>"
+lang_name     "<U004B><U0069><U0073><U0077><U0061><U0068><U0069><U006C><U0069>"
+lang_ab       "<U0073><U0077>"
+lang_term     "<U0073><U0077><U0061>"
+lang_lib      "<U0073><U0077><U0061>"
+END LC_ADDRESS
+
+LC_NAME
+% FIXME!
+copy "en_US"
+END LC_NAME
diff -urN glibc-2.6.org/localedata/locales/sw_UG glibc-2.6/localedata/locales/sw_UG
--- glibc-2.6.org/localedata/locales/sw_UG	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.6/localedata/locales/sw_UG	2007-05-18 12:14:58.623621321 +0200
@@ -0,0 +1,111 @@
+comment_char %
+escape_char  /
+%
+% Swahili Language Locale for Uganda
+% Source: Based on data from Debian/Mandriva
+% Contact: 
+% Email: ankry at mif dot pg dot gda dot pl
+% Language: sw
+% Territory: Uganda
+% Revision: 0.2
+% Date: 2004-08-01
+% Users: general
+% Repertoiremap: mnemonic,ds
+% Charset: UTF-8
+% Distribution and use is free, also
+% for commercial purposes.
+
+LC_IDENTIFICATION
+title      "Swahili locale for Uganda"
+source     "Based on data from Debian/Mandriva"
+address    ""
+contact    ""
+email      ""
+tel        ""
+fax        ""
+language   "Swahili"
+territory  "Uganda"
+revision   "0.3"
+date       "2005-10-14"
+%
+category  "i18n:1997";LC_IDENTIFICATION
+category  "i18n:1997";LC_CTYPE
+category  "i18n:1997";LC_COLLATE
+category  "i18n:1997";LC_TIME
+category  "i18n:1997";LC_NUMERIC
+category  "i18n:1997";LC_MONETARY
+category  "i18n:1997";LC_MESSAGES
+category  "i18n:1997";LC_PAPER
+category  "i18n:1997";LC_NAME
+category  "i18n:1997";LC_ADDRESS
+category  "i18n:1997";LC_TELEPHONE
+END LC_IDENTIFICATION
+
+%%%%%%%%%%%%%%%%
+% TODO: check LC_MONETARY, LC_NUMERIC, LC_MESSAGES, LC_COLLATE
+% 	check time and date formats
+%%%%%%%%%%%%%%%%
+
+LC_COLLATE
+copy "sw_KE"
+END LC_COLLATE
+
+LC_CTYPE
+copy "sw_KE"
+END LC_CTYPE
+
+LC_MONETARY
+copy "lg_UG"
+END LC_MONETARY
+
+LC_NUMERIC
+copy "lg_UG"
+END LC_NUMERIC
+
+LC_TELEPHONE
+copy "lg_UG"
+END LC_TELEPHONE
+
+LC_TIME
+copy "sw_KE"
+END LC_TIME
+
+LC_MESSAGES
+copy "sw_KE"
+END LC_MESSAGES
+
+% Default paper: A4
+LC_PAPER
+height 297
+width 210
+END LC_PAPER
+
+% Metric System (ISO 1000)
+LC_MEASUREMENT
+measurement 1
+END LC_MEASUREMENT
+
+LC_ADDRESS
+%
+% This is the ISO_IEC TR14652 Locale definition for the LC_ADDRESS category.
+%
+postal_fmt    "<U0025><U007A><U0025><U0063><U0025><U0054><U0025><U0073>/
+<U0025><U0062><U0025><U0065><U0025><U0072>"
+%
+% FIXME: Uganda in Swahili
+%country_name  ""
+country_post  "<U0055><U0047>"
+country_ab2   "<U0055><U0047>"
+country_ab3   "<U0055><U0047><U0041>"
+country_num   800
+country_car   "<U0045><U0041><U0055>"
+country_isbn  "<U0039><U0039><U0037><U0030>"
+lang_name     "<U004B><U0069><U0073><U0077><U0061><U0068><U0069><U006C><U0069>"
+lang_ab       "<U0073><U0077>"
+lang_term     "<U0073><U0077><U0061>"
+lang_lib      "<U0073><U0077><U0061>"
+END LC_ADDRESS
+
+LC_NAME
+copy "sw_KE"
+END LC_NAME
--- glibc-2.8/localedata/SUPPORTED.orig	2008-04-16 20:55:48.000000000 +0200
+++ glibc-2.8/localedata/SUPPORTED	2008-10-08 19:06:05.514417473 +0200
@@ -47,12 +47,15 @@
 ar_YE.UTF-8/UTF-8 \
 ar_YE/ISO-8859-6 \
 az_AZ.UTF-8/UTF-8 \
+az_AZ/UTF-8 \
 as_IN.UTF-8/UTF-8 \
 ast_ES.UTF-8/UTF-8 \
 ast_ES/ISO-8859-15 \
 be_BY.UTF-8/UTF-8 \
 be_BY/CP1251 \
 be_BY@latin/UTF-8 \
+be_BY@alternative.UTF-8/UTF-8 \
+be_BY@alternative/CP1251 \
 ber_DZ/UTF-8 \
 ber_MA/UTF-8 \
 bg_BG.UTF-8/UTF-8 \
@@ -132,6 +136,8 @@
 en_ZA/ISO-8859-1 \
 en_ZW.UTF-8/UTF-8 \
 en_ZW/ISO-8859-1 \
+eo/ISO-8859-3 \
+eo.UTF-8/UTF-8 \
 es_AR.UTF-8/UTF-8 \
 es_AR/ISO-8859-1 \
 es_BO.UTF-8/UTF-8 \
@@ -179,10 +185,14 @@
 eu_ES.UTF-8/UTF-8 \
 eu_ES/ISO-8859-1 \
 eu_ES@euro/ISO-8859-15 \
+eu_FR.UTF-8/UTF-8 \
+eu_FR/ISO-8859-1 \
+eu_FR@euro/ISO-8859-15 \
 fa_IR/UTF-8 \
 fi_FI.UTF-8/UTF-8 \
 fi_FI/ISO-8859-1 \
 fi_FI@euro/ISO-8859-15 \
+fil_PH.ISO-8859-1/ISO-8859-1 \
 fil_PH/UTF-8 \
 fo_FO.UTF-8/UTF-8 \
 fo_FO/ISO-8859-1 \
@@ -199,8 +209,14 @@
 fr_LU.UTF-8/UTF-8 \
 fr_LU/ISO-8859-1 \
 fr_LU@euro/ISO-8859-15 \
+fur_IT.ISO-8859-1/ISO-8859-1 \
+fur_IT.ISO-8859-15/ISO-8859-15 \
 fur_IT/UTF-8 \
+fy_NL.ISO-8859-1/ISO-8859-1 \
+fy_NL.ISO-8859-15/ISO-8859-15 \
 fy_NL/UTF-8 \
+fy_DE.ISO-8859-1/ISO-8859-1 \
+fy_DE.ISO-8859-15/ISO-8859-15 \
 fy_DE/UTF-8 \
 ga_IE.UTF-8/UTF-8 \
 ga_IE/ISO-8859-1 \
@@ -236,17 +236,19 @@
 ha_NG/UTF-8 \
 he_IL.UTF-8/UTF-8 \
 he_IL/ISO-8859-8 \
+hi_IN.ISCII-DEV/ISCII-DEV \
 hi_IN/UTF-8 \
 hne_IN/UTF-8 \
 hr_HR.UTF-8/UTF-8 \
 hr_HR/ISO-8859-2 \
 hsb_DE/ISO-8859-2 \
 hsb_DE.UTF-8/UTF-8 \
 ht_HT/UTF-8 \
+hu_HU.ISO-8859-16/ISO-8859-16 \
 hu_HU.UTF-8/UTF-8 \
 hu_HU/ISO-8859-2 \
-hy_AM/UTF-8 \
 hy_AM.ARMSCII-8/ARMSCII-8 \
+hy_AM/UTF-8 \
 id_ID.UTF-8/UTF-8 \
 id_ID/ISO-8859-1 \
 ig_NG/UTF-8 \
@@ -241,10 +260,9 @@
 it_IT/ISO-8859-1 \
 it_IT@euro/ISO-8859-15 \
 iu_CA/UTF-8 \
-iw_IL.UTF-8/UTF-8 \
-iw_IL/ISO-8859-8 \
 ja_JP.EUC-JP/EUC-JP \
 ja_JP.UTF-8/UTF-8 \
+ka_GE.GEORGIAN-ACADEMY/GEORGIAN-ACADEMY \
 ka_GE.UTF-8/UTF-8 \
 ka_GE/GEORGIAN-PS \
 kk_KZ.UTF-8/UTF-8 \
@@ -278,11 +278,17 @@
 ku_TR/ISO-8859-9 \
 kw_GB.UTF-8/UTF-8 \
 kw_GB/ISO-8859-1 \
+ky_KG.KOI8-K/KOI8-K \
 ky_KG/UTF-8 \
 lg_UG.UTF-8/UTF-8 \
 lg_UG/ISO-8859-10 \
+li_BE.ISO-8859-1/ISO-8859-1 \
+li_BE.ISO-8859-15/ISO-8859-15 \
 li_BE/UTF-8 \
+li_NL.ISO-8859-1/ISO-8859-1 \
+li_NL.ISO-8859-15/ISO-8859-15 \
 li_NL/UTF-8 \
+lo_LA.MULELAO-1/MULELAO-1 \
 lo_LA/UTF-8 \
 lt_LT.UTF-8/UTF-8 \
 lt_LT/ISO-8859-13 \
@@ -310,7 +310,11 @@
 mt_MT/ISO-8859-3 \
 nb_NO.UTF-8/UTF-8 \
 nb_NO/ISO-8859-1 \
+nds_DE.ISO-8859-1/ISO-8859-1 \
+nds_DE.ISO-8859-15/ISO-8859-15 \
 nds_DE/UTF-8 \
+nds_DE.ISO-8859-1@traditional/ISO-8859-1 \
+nds_DE@traditional/UTF-8 \
 nds_NL/UTF-8 \
 ne_NP/UTF-8 \
 nl_AW/UTF-8 \
@@ -296,24 +326,33 @@
 nl_NL@euro/ISO-8859-15 \
 nn_NO.UTF-8/UTF-8 \
 nn_NO/ISO-8859-1 \
+nr_ZA.ISO-8859-1/ISO-8859-1 \
 nr_ZA/UTF-8 \
+nso_ZA.ISO-8859-10/ISO-8859-10 \
 nso_ZA/UTF-8 \
 oc_FR.UTF-8/UTF-8 \
 oc_FR/ISO-8859-1 \
+oc_FR@euro/ISO-8859-15 \
 om_ET/UTF-8 \
 om_KE.UTF-8/UTF-8 \
 om_KE/ISO-8859-1 \
 or_IN/UTF-8 \
 pa_IN/UTF-8 \
 pa_PK/UTF-8 \
+pap_AN.ISO-8859-1/ISO-8859-1 \
+pap_AN.ISO-8859-15/ISO-8859-15 \
 pap_AN/UTF-8 \
+pl_PL.ISO-8859-16/ISO-8859-16 \
 pl_PL.UTF-8/UTF-8 \
 pl_PL/ISO-8859-2 \
 ps_AF/UTF-8 \
 pt_BR.UTF-8/UTF-8 \
 pt_BR/ISO-8859-1 \
 pt_PT.UTF-8/UTF-8 \
 pt_PT/ISO-8859-1 \
 pt_PT@euro/ISO-8859-15 \
+rm_CH.UTF-8/UTF-8 \
+rm_CH/ISO-8859-1 \
+ro_RO.ISO-8859-16/ISO-8859-16 \
 ro_RO.UTF-8/UTF-8 \
 ro_RO/ISO-8859-2 \
@@ -361,17 +361,21 @@
 ru_RU/ISO-8859-5 \
 ru_UA.UTF-8/UTF-8 \
 ru_UA/KOI8-U \
+rw_RW.ISO-8859-1/ISO-8859-1 \
 rw_RW/UTF-8 \
 sa_IN/UTF-8 \
+sc_IT.ISO-8859-1/ISO-8859-1 \
+sc_IT.ISO-8859-15/ISO-8859-15 \
 sc_IT/UTF-8 \
 sd_IN/UTF-8 \
 sd_IN@devanagari/UTF-8 \
 se_NO/UTF-8 \
 shs_CA/UTF-8 \
 si_LK/UTF-8 \
 sid_ET/UTF-8 \
 sk_SK.UTF-8/UTF-8 \
 sk_SK/ISO-8859-2 \
+sl_SI.ISO-8859-16/ISO-8859-16 \
 sl_SI.UTF-8/UTF-8 \
 sl_SI/ISO-8859-2 \
 so_DJ.UTF-8/UTF-8 \
@@ -341,9 +387,13 @@
 sq_AL.UTF-8/UTF-8 \
 sq_AL/ISO-8859-1 \
 sq_MK/UTF-8 \
+sr_ME.ISO-8859-5/ISO-8859-5 \
 sr_ME/UTF-8 \
+sr_RS.ISO-8859-5/ISO-8859-5 \
 sr_RS/UTF-8 \
+sr_RS.ISO-8859-2@latin/ISO-8859-2 \
 sr_RS@latin/UTF-8 \
+ss_ZA.ISO-8859-1/ISO-8859-1 \
 ss_ZA/UTF-8 \
 st_ZA.UTF-8/UTF-8 \
 st_ZA/ISO-8859-1 \
@@ -352,10 +402,17 @@
 sv_FI@euro/ISO-8859-15 \
 sv_SE.UTF-8/UTF-8 \
 sv_SE/ISO-8859-1 \
+sw_KE.UTF-8/UTF-8 \
+sw_KE/ISO-8859-1 \
+sw_UG.UTF-8/UTF-8 \
+sw_UG.ISO-8859-10/ISO-8859-10 \
+sw_UG/ISO-8859-1 \
 ta_IN/UTF-8 \
 te_IN/UTF-8 \
+tg_TJ.KOI8-K/KOI8-K \
 tg_TJ.UTF-8/UTF-8 \
 tg_TJ/KOI8-T \
+th_TH.ISO-8859-11/ISO-8859-11 \
 th_TH.UTF-8/UTF-8 \
 th_TH/TIS-620 \
 ti_ER/UTF-8 \
@@ -364,18 +421,24 @@
 tk_TM/UTF-8 \
 tl_PH.UTF-8/UTF-8 \
 tl_PH/ISO-8859-1 \
+tn_ZA.ISO-8859-10/ISO-8859-10 \
 tn_ZA/UTF-8 \
 tr_CY.UTF-8/UTF-8 \
 tr_CY/ISO-8859-9 \
 tr_TR.UTF-8/UTF-8 \
 tr_TR/ISO-8859-9 \
+ts_ZA.ISO-8859-1/ISO-8859-1 \
 ts_ZA/UTF-8 \
+tt_RU.KOI8-K/KOI8-K \
 tt_RU.UTF-8/UTF-8 \
 tt_RU.UTF-8@iqtelif/UTF-8 \
+tt_RU/TATAR-CYR \
 ug_CN/UTF-8 \
 uk_UA.UTF-8/UTF-8 \
 uk_UA/KOI8-U \
+ur_PK.CP1256/CP1256 \
 ur_PK/UTF-8 \
+uz_UZ.UTF-8/UTF-8 \
 uz_UZ/ISO-8859-1 \
 uz_UZ@cyrillic/UTF-8 \
 ve_ZA/UTF-8 \
@@ -384,6 +447,7 @@
 wa_BE/ISO-8859-1 \
 wa_BE@euro/ISO-8859-15 \
 wa_BE.UTF-8/UTF-8 \
+wal_ET/UTF-8 \
 wo_SN/UTF-8 \
 xh_ZA.UTF-8/UTF-8 \
 xh_ZA/ISO-8859-1 \
