diff -Naur glibc-2.2.3/sysdeps/unix/sysv/linux/paths.h glibc-2.2.3-p/sysdeps/unix/sysv/linux/paths.h
--- glibc-2.2.3/sysdeps/unix/sysv/linux/paths.h	Thu Nov 16 20:12:05 2000
+++ glibc-2.2.3-p/sysdeps/unix/sysv/linux/paths.h	Tue Jul  3 15:00:49 2001
@@ -55,13 +55,13 @@
 #define	_PATH_NOLOGIN	"/etc/nologin"
 #define	_PATH_PRESERVE	"/var/lib"
 #define	_PATH_RWHODIR	"/var/spool/rwho"
-#define	_PATH_SENDMAIL	"/usr/sbin/sendmail"
+#define	_PATH_SENDMAIL	"/usr/lib/sendmail"
 #define	_PATH_SHADOW	"/etc/shadow"
 #define	_PATH_SHELLS	"/etc/shells"
 #define	_PATH_TTY	"/dev/tty"
 #define	_PATH_UNIX	"/boot/vmlinux"
 #define _PATH_UTMP	"/var/run/utmp"
-#define	_PATH_VI	"/usr/bin/vi"
+#define	_PATH_VI	"/bin/vi"
 #define _PATH_WTMP	"/var/log/wtmp"
 
 /* Provide trailing slash, since mostly used for building pathnames. */
--- glibc-2.3.4/nscd/nscd.h.orig	2004-10-15 23:13:19.000000000 +0200
+++ glibc-2.3.4/nscd/nscd.h	2005-02-20 16:26:37.755491792 +0100
@@ -91,10 +91,10 @@
 
 
 /* Paths of the file for the persistent storage.  */
-#define _PATH_NSCD_PASSWD_DB	"/var/db/nscd/passwd"
-#define _PATH_NSCD_GROUP_DB	"/var/db/nscd/group"
-#define _PATH_NSCD_HOSTS_DB	"/var/db/nscd/hosts"
-#define _PATH_NSCD_SERVICES_DB	"/var/db/nscd/services"
+#define _PATH_NSCD_PASSWD_DB	"/var/lib/nscd/passwd"
+#define _PATH_NSCD_GROUP_DB	"/var/lib/nscd/group"
+#define _PATH_NSCD_HOSTS_DB	"/var/lib/nscd/hosts"
+#define _PATH_NSCD_SERVICES_DB	"/var/lib/nscd/services"
 
 /* Path used when not using persistent storage.  */
 #define _PATH_NSCD_XYZ_DB_TMP	"/var/run/nscd/dbXXXXXX"

