diff -Nru glibc-2.1/manual/libc.texinfo glibc-2.1.new/manual/libc.texinfo
--- glibc-2.1/manual/libc.texinfo	Tue Jan 12 18:22:16 1999
+++ glibc-2.1.new/manual/libc.texinfo	Sat Feb 27 02:40:12 1999
@@ -5,9 +5,9 @@
 @c setchapternewpage odd
 
 @comment Tell install-info what to do.
-@dircategory Software libraries
+@dircategory Libraries:
 @direntry
-* Libc: (libc).                 C library.
+* Libc: (libc).				C library.
 @end direntry
 @include dir-add.texi
 

