--- glibc-2.8/nscd/nscd.conf.org	2008-04-16 21:11:39.024545404 +0200
+++ glibc-2.8/nscd/nscd.conf	2008-04-16 21:12:10.936070956 +0200
@@ -30,11 +30,11 @@
 #
 
 
-#	logfile			/var/log/nscd.log
-#	threads			4
-#	max-threads		32
-#	server-user		nobody
-#	stat-user		somebody
+	logfile			/var/log/nscd
+	threads			4
+	max-threads		32
+	server-user		nscd
+	stat-user		root
 	debug-level		0
 #	reload-count		5
 	paranoia		no
