--- libc/time/tzfile.c.orig	2004-03-20 01:37:50.000000000 +0100
+++ libc/time/tzfile.c	2004-06-29 17:40:15.000000000 +0200
@@ -313,7 +313,9 @@
     {
       /* This should only happen if there are no transition rules.
 	 In this case there should be only one single type.  */
-      assert (num_types == 1);
+      if(__builtin_expect((num_types != 1), 0))
+	goto ret_free_transitions;
+      /* assert (num_types == 1);  -- disabled, don't crash */
       __tzname[0] = __tzstring (zone_names);
     }
   if (__tzname[1] == NULL)
