--- glibc-2.3.2/Makeconfig.orig	Mon Jan  6 06:31:36 2003
+++ glibc-2.3.2/Makeconfig	Fri May  9 10:05:02 2003
@@ -691,8 +691,8 @@
 object-suffixes += .og
 CPPFLAGS-.og = $(pic-default)
 CFLAGS-.og = -g
-CFLAGS-.o = $(filter %frame-pointer,$(+cflags)) -g0 -O99 -fomit-frame-pointer -D__USE_STRING_INLINES
-CFLAGS-.os += -g0 -O99 -fomit-frame-pointer -D__USE_STRING_INLINES
+CFLAGS-.o = $(filter %frame-pointer,$(+cflags)) -fomit-frame-pointer -D__USE_STRING_INLINES
+CFLAGS-.os += -fomit-frame-pointer -D__USE_STRING_INLINES
 libtype.og = lib%_g.a
 endif
 
