--- hugin-0.7.0/src/hugin_base/CMakeLists.txt~	2008-09-03 21:59:25.000000000 +0100
+++ hugin-0.7.0/src/hugin_base/CMakeLists.txt	2008-09-09 18:56:04.000000000 +0100
@@ -58,6 +58,7 @@
 IF (${HUGIN_SHARED_LIBS})
   add_library(huginbase SHARED ${HUGIN_BASE_SRC})
   set_target_properties(huginbase PROPERTIES VERSION ${HUGIN_LIB_VERSION})
+  target_link_libraries(huginbase huginvigraimpex ${PANO_LIBRARIES} ${Boost_thread_LIBRARIES} ${EXIV2_LIBRARIES})
   install(TARGETS huginbase DESTINATION ${LIBDIR})
 ELSE (${HUGIN_SHARED_LIBS})
   add_library(huginbase STATIC ${HUGIN_BASE_SRC})
--- hugin-0.7.0/src/foreign/vigra/vigra_impex/CMakeLists.txt~	2008-09-03 21:59:24.000000000 +0100
+++ hugin-0.7.0/src/foreign/vigra/vigra_impex/CMakeLists.txt	2008-09-09 18:56:31.000000000 +0100
@@ -5,6 +5,7 @@
 IF (${HUGIN_SHARED_LIBS})
   add_library(huginvigraimpex SHARED ${VIGRA_IMPEX_SRC})
   set_target_properties(huginvigraimpex PROPERTIES VERSION ${HUGIN_LIB_VERSION})
+  target_link_libraries(huginvigraimpex ${OPENEXR_LIBRARIES} ${JPEG_LIBRARIES} ${PNG_LIBRARIES} ${TIFF_LIBRARIES})
   install(TARGETS huginvigraimpex DESTINATION ${LIBDIR})
 ELSE (${HUGIN_SHARED_LIBS})
   add_library(huginvigraimpex STATIC ${VIGRA_IMPEX_SRC})
