diff -uNr js-1.8.5.old/js/src/Makefile.in js-1.8.5/js/src/Makefile.in
--- js-1.8.5.old/js/src/Makefile.in	2011-03-31 21:08:36.000000000 +0200
+++ js-1.8.5/js/src/Makefile.in	2012-01-18 15:03:30.706388915 +0100
@@ -888,8 +888,8 @@
 ifeq (,$(HOST_BIN_SUFFIX))
 	mv -f $(SHLIB_ANY_VER) $(SHLIB_EXACT_VER)
 	@[ ! -h $(SHLIB_ABI_VER) ] || rm -f $(SHLIB_ABI_VER)
-	ln -s $(SHLIB_EXACT_VER) $(SHLIB_ABI_VER)
-	ln -s $(SHLIB_ABI_VER) $(SHLIB_ANY_VER)
+	ln -s $(notdir $(SHLIB_EXACT_VER)) $(SHLIB_ABI_VER)
+	ln -s $(notdir $(SHLIB_ABI_VER)) $(SHLIB_ANY_VER)
 endif
 endif
 ifneq (,$(IMPORT_LIBRARY))
