diff -urN cups-1.2.0.org/conf/cupsd.conf.in cups-1.2.0/conf/cupsd.conf.in
--- cups-1.2.0.org/conf/cupsd.conf.in	2006-04-23 23:46:38.000000000 +0200
+++ cups-1.2.0/conf/cupsd.conf.in	2006-05-12 08:17:48.946318536 +0200
@@ -6,6 +6,9 @@
 #   file.
 #
 
+MaxLogSize 0
+Printcap /etc/printcap
+
 # Log general information in error_log - change "@CUPS_LOG_LEVEL@" to "debug"
 # for troubleshooting...
 LogLevel @CUPS_LOG_LEVEL@
