diff -urN tcl8.5a2.org/unix/configure.in tcl8.5a2/unix/configure.in
--- tcl8.5a2.org/unix/configure.in	2004-12-08 00:16:06.000000000 +0100
+++ tcl8.5a2/unix/configure.in	2005-04-04 13:19:13.000000000 +0200
@@ -436,7 +436,11 @@
 
 TCL_UNSHARED_LIB_SUFFIX=${UNSHARED_LIB_SUFFIX}
 TCL_SHARED_LIB_SUFFIX=${SHARED_LIB_SUFFIX}
-eval "TCL_LIB_FILE=libtcl${LIB_SUFFIX}"
+TCL_SOMAJOR=".0"
+TCL_SOVERSION="${TCL_SOMAJOR}.0"
+eval "TCL_LIB_FILE=libtcl${LIB_SUFFIX}${TCL_SOVERSION}"
+eval "TCL_LIB_SONAME=libtcl${LIB_SUFFIX}${TCL_SOMAJOR}"
+AC_SUBST(TCL_LIB_SONAME)
 
 SC_ENABLE_FRAMEWORK
 
diff -urN tcl8.5a2.org/unix/Makefile.in tcl8.5a2/unix/Makefile.in
--- tcl8.5a2.org/unix/Makefile.in	2004-12-08 00:16:06.000000000 +0100
+++ tcl8.5a2/unix/Makefile.in	2005-04-04 13:20:39.000000000 +0200
@@ -459,7 +459,9 @@
 # a shared library or non-shared library for Tcl.
 ${LIB_FILE}: ${OBJS} ${STUB_LIB_FILE}
 	rm -f $@
-	@MAKE_LIB@
+	@MAKE_LIB@ -Wl,-soname,@TCL_LIB_SONAME@
+	-ln -sf ${TCL_LIB_FILE} libtcl${VERSION}@SHLIB_SUFFIX@
+	-ln -sf ${TCL_LIB_FILE} libtcl${VERSION}@SHLIB_SUFFIX@.0
 
 ${STUB_LIB_FILE}: ${STUB_LIB_OBJS}
 	rm -f $@

