--- cronie-1.4.2/anacron/global.h~	2009-09-25 09:23:18.000000000 +0300
+++ cronie-1.4.2/anacron/global.h	2009-11-17 08:08:34.079251474 +0200
@@ -33,7 +33,7 @@
 #define DEBUG_LEVEL LOG_DEBUG     /* only used when DEBUG is defined */
 
 /* Mail interface.  (All MTAs should supply this command) */
-#define SENDMAIL "/usr/sbin/sendmail"
+#define SENDMAIL "/usr/lib/sendmail"
 
 /* End of user-configurable section */
 
--- cronie-1.4.2/anacron/runjob.c~	2009-09-25 09:23:18.000000000 +0300
+++ cronie-1.4.2/anacron/runjob.c	2009-11-17 08:10:16.259239987 +0200
@@ -195,7 +195,7 @@
 	/* fdflags = fcntl(0, F_GETFL); fdflags &= ~O_APPEND; */
 	/* fcntl(0, F_SETFL, fdflags ); */
 
-	/* Here, I basically mirrored the way /usr/sbin/sendmail is called
+	/* Here, I basically mirrored the way /usr/lib/sendmail is called
 	 * by cron on a Debian system, except for the "-oem" and "-or0s"
 	 * options, which don't seem to be appropriate here.
 	 * Hopefully, this will keep all the MTAs happy. */
--- cronie-1.4.2/configure.ac~	2009-11-17 08:07:30.000000000 +0200
+++ cronie-1.4.2/configure.ac	2009-11-17 08:11:22.722566791 +0200
@@ -189,7 +189,7 @@
 
 AC_DEFINE(DEBUGGING,1,[Code will be built with debug info.])
 
-AC_DEFINE(MAILARG,"/usr/sbin/sendmail",[There will be path to sendmail.])
+AC_DEFINE(MAILARG,"/usr/lib/sendmail",[There will be path to sendmail.])
 
 AC_DEFINE(MAILFMT,"%s -FCronDaemon -i -odi -oem -oi -t -f %s",
 [-i    = don't terminate on "." by itself
