diff -urN gobject-introspection-0.6.2/configure.ac gobject-introspection-0.6.2.new/configure.ac
--- gobject-introspection-0.6.2/configure.ac	2008-11-25 23:57:07.000000000 +0100
+++ gobject-introspection-0.6.2.new/configure.ac	2009-01-24 20:32:12.000000000 +0100
@@ -36,6 +36,9 @@
 AC_PROG_LIBTOOL
 PKG_PROG_PKG_CONFIG
 
+dnl when using libtool 2.x create libtool early, because it's used in configure
+m4_ifdef([LT_OUTPUT], [LT_OUTPUT])
+
 AC_PROG_LEX
 if test "$LEX" = :; then
 	AC_MSG_ERROR([flex not found but required])
