--- expat-2.0.0/conftools/expat.m4.orig	2001-11-09 05:55:33.000000000 +0100
+++ expat-2.0.0/conftools/expat.m4	2006-01-15 15:08:05.610539500 +0100
@@ -7,7 +7,7 @@
 dnl In addition, an Automake conditional EXPAT_INSTALLED is set accordingly.
 dnl This is necessary to adapt a whole lot of packages that have expat
 dnl bundled as a static library.
-AC_DEFUN(AM_WITH_EXPAT,
+AC_DEFUN([AM_WITH_EXPAT],
 [ AC_ARG_WITH(expat,
 	      [  --with-expat=PREFIX     Use system Expat library],
 	      , with_expat=no)
