diff -aurN xcb-util-0.3.9.orig/src/Makefile.am xcb-util-0.3.9/src/Makefile.am
--- xcb-util-0.3.9.orig/src/Makefile.am	2012-06-03 17:16:59.599691723 +0200
+++ xcb-util-0.3.9/src/Makefile.am	2012-06-03 17:17:25.003374131 +0200
@@ -4,7 +4,7 @@
 AM_CFLAGS = $(CWARNFLAGS)
 
 libxcb_util_la_LIBADD = $(XCB_LIBS)
-libxcb_util_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libxcb_util_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 libxcb_util_la_SOURCES = \
 	atoms.c		\
 	event.c		\
