diff -aurN jack-1.9.8.orig/common/wscript jack-1.9.8/common/wscript
--- jack-1.9.8.orig/common/wscript	2011-10-20 10:16:56.372293010 +0200
+++ jack-1.9.8/common/wscript	2011-10-20 10:21:29.726875607 +0200
@@ -191,7 +191,7 @@
         netlib.includes = includes
         netlib.name         = 'netlib'
         netlib.target       = 'jacknet'
-        netlib.uselib = ['SAMPLERATE', 'CELT']
+        netlib.uselib = ['SAMPLERATE', 'CELT', 'PTHREAD' , 'RT']
         netlib.install_path = '${LIBDIR}'
         netlib.source = [
             'JackNetAPI.cpp',
