--- coregrind/m_debuginfo/readelf.c~	2011-10-26 23:24:44.000000000 +0200
+++ coregrind/m_debuginfo/readelf.c	2011-12-10 16:02:00.557983343 +0100
@@ -1198,7 +1198,9 @@
          VG_(sprintf)(debugpath, "%s/.debug/%s", objdir, debugname);
          if ((addr = open_debug_file(debugpath, NULL, crc, &size)) == 0) {
             VG_(sprintf)(debugpath, "/usr/lib/debug%s/%s", objdir, debugname);
-            addr = open_debug_file(debugpath, NULL, crc, &size);
+            if ((addr = open_debug_file(debugpath, NULL, crc, &size)) == 0) {
+                addr = open_debug_file(debugname, NULL, crc, size);
+            }
          }
       }
 
