diff -aurN shotwell-0.12.1.orig/Makefile shotwell-0.12.1/Makefile
--- shotwell-0.12.1.orig/Makefile	2012-04-07 12:33:35.361408718 +0200
+++ shotwell-0.12.1/Makefile	2012-04-07 12:39:34.187922771 +0200
@@ -265,8 +265,6 @@
 	posix \
 	LConv
 
-# libraw is not listed (see note below); when libraw-config is no longer needed, it should be
-# added to this list
 EXT_PKGS = \
 	atk \
 	gdk-3.0 \
@@ -284,6 +282,7 @@
 	libexif \
 	libgphoto2 \
 	libsoup-2.4 \
+	libraw \
 	libxml-2.0 \
 	sqlite3 \
 	unique-3.0 \
@@ -300,11 +299,6 @@
 
 DIRECT_LIBS =
 
-LIBRAW_PKG = \
-	libraw
-
-# libraw is handled separately (see note below); when libraw-config is no longer needed, the version
-# should be added to this list
 EXT_PKG_VERSIONS = \
 	gee-1.0 >= 0.5.0 \
 	gexiv2 >= 0.3.92 \
@@ -320,6 +314,7 @@
 	libexif >= 0.6.16 \
 	libgphoto2 >= 2.4.2 \
 	libsoup-2.4 >= 2.26.0 \
+	libraw >= 0.13.4 \
 	libxml-2.0 >= 2.6.32 \
 	sqlite3 >= 3.5.9 \
 	unique-3.0 >= 3.0.0 \
@@ -331,10 +326,7 @@
 endif
 DIRECT_LIBS_VERSIONS =
 
-LIBRAW_VERSION = \
-	0.9.0
-
-VALA_PKGS = $(EXT_PKGS) $(LOCAL_PKGS) $(LIBRAW_PKG)
+VALA_PKGS = $(EXT_PKGS) $(LOCAL_PKGS)
 
 ifndef BUILD_DIR
 BUILD_DIR=src
@@ -425,21 +417,14 @@
 CFLAGS = -O0 -g -pipe
 PLUGIN_CFLAGS = -O0 -g -pipe
 else
-CFLAGS = -O2 -g -pipe
-PLUGIN_CFLAGS = -O2 -g -pipe
+CFLAGS = $(OPTFLAGS)
+PLUGIN_CFLAGS = $(OPTFLAGS)
 endif
 endif
 
 CFLAGS += $(REQUIRED_CFLAGS)
 PLUGIN_CFLAGS += $(REQUIRED_CFLAGS)
 
-# Required for gudev-1.0
-CFLAGS += -DG_UDEV_API_IS_SUBJECT_TO_CHANGE
-
-# Packaged libraw is not widely available, so we must fake what would be in its .pc file
-# if not available.
-LIBRAW_CONFIG=./libraw-config
-
 define check_valac_version
 	@ ./chkver min $(VALAC_VERSION) $(MIN_VALAC_VERSION) || ( echo 'Shotwell requires Vala compiler $(MIN_VALAC_VERSION) or greater.  You are running' $(VALAC_VERSION) '\b.'; exit 1 )
 	$(if $(MAX_VALAC_VERSION),\
@@ -683,10 +668,10 @@
 	@
 
 $(EXPANDED_OBJ_FILES): %.o: %.c $(CONFIG_IN) Makefile
-	$(CC) -c $(VALA_CFLAGS) `$(LIBRAW_CONFIG) --cflags` $(CFLAGS) -o $@ $<
+	$(CC) -c $(VALA_CFLAGS) $(CFLAGS) -o $@ $<
 
 $(PROGRAM): $(EXPANDED_OBJ_FILES) $(RESOURCES) $(LANG_STAMP) $(THUMBNAILER_BIN)
-	$(CC) $(EXPANDED_OBJ_FILES) $(CFLAGS) $(LDFLAGS) $(RESOURCES) $(VALA_LDFLAGS) `$(LIBRAW_CONFIG) --libs` $(EXPORT_FLAGS) -o $@
+	$(CC) $(EXPANDED_OBJ_FILES) $(CFLAGS) $(LDFLAGS) $(RESOURCES) $(VALA_LDFLAGS) $(EXPORT_FLAGS) -o $@
 	glib-compile-schemas misc
 
 $(THUMBNAILER_BIN): $(EXPANDED_THUMBNAILER_SRC_FILES)
@@ -715,7 +700,7 @@
 glade: lib$(PROGRAM).so
 
 lib$(PROGRAM).so: $(EXPANDED_OBJ_FILES) $(RESOURCES) $(LANG_STAMP)
-	$(CC) $(EXPANDED_OBJ_FILES) $(CFLAGS) $(LDFLAGS) $(RESOURCES) $(VALA_LDFLAGS) `$(LIBRAW_CONFIG) --libs` $(EXPORT_FLAGS) -shared -o $@
+	$(CC) $(EXPANDED_OBJ_FILES) $(CFLAGS) $(LDFLAGS) $(RESOURCES) $(VALA_LDFLAGS) $(EXPORT_FLAGS) -shared -o $@
 
 .PHONY: pkgcheck
 pkgcheck:
@@ -726,7 +711,5 @@
 ifdef EXT_PKGS
 	@pkg-config --print-errors --exists $(EXT_PKGS) $(DIRECT_LIBS_VERSIONS)
 endif
-# Check for libraw manually
-	@$(LIBRAW_CONFIG) --exists=$(LIBRAW_VERSION)
 endif
 	@ type msgfmt > /dev/null || ( echo 'msgfmt (usually found in the gettext package) is missing and is required to build Shotwell. ' ; exit 1 )
diff -aurN shotwell-0.12.1.orig/plugins/Makefile.plugin.mk shotwell-0.12.1/plugins/Makefile.plugin.mk
--- shotwell-0.12.1.orig/plugins/Makefile.plugin.mk	2012-04-07 12:33:35.358408757 +0200
+++ shotwell-0.12.1/plugins/Makefile.plugin.mk	2012-04-07 12:41:17.488631334 +0200
@@ -34,7 +34,7 @@
 CFILES := $(notdir $(SRC_FILES:.vala=.c))
 OFILES := $(notdir $(SRC_FILES:.vala=.o))
 
-CFLAGS := `pkg-config --print-errors --cflags $(EXT_PKGS) $(PLUGIN_PKGS)` -nostdlib -export-dynamic $(PLUGIN_CFLAGS)
+CFLAGS := `pkg-config --print-errors --cflags $(EXT_PKGS) $(PLUGIN_PKGS)` -nostdlib -export-dynamic $(PLUGIN_CFLAGS) $(OPTFLAGS)
 LIBS := `pkg-config --print-errors --libs $(EXT_PKGS) $(PLUGIN_PKGS)`
 DEFINES := -D_VERSION='"$(PLUGINS_VERSION)"' -DGETTEXT_PACKAGE='"shotwell"'
 
