diff -aurN metacity-2.26.0.orig/src/metacity.schemas.in.in metacity-2.26.0/src/metacity.schemas.in.in
--- metacity-2.26.0.orig/src/metacity.schemas.in.in	2009-03-24 16:37:35.000000000 +0100
+++ metacity-2.26.0/src/metacity.schemas.in.in	2009-03-24 17:59:48.000000000 +0100
@@ -247,7 +247,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Shiki-Colors-Easy-Metacity</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -278,7 +278,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>Sans Bold 9</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
