diff -aurN jconvolver-0.9.1.orig/source/Makefile jconvolver-0.9.1/source/Makefile
--- jconvolver-0.9.1.orig/source/Makefile	2011-10-20 11:36:31.862591166 +0200
+++ jconvolver-0.9.1/source/Makefile	2011-10-20 17:58:48.682841452 +0200
@@ -24,10 +24,9 @@
 LIBDIR = lib$(SUFFIX)
 DISTDIR = jconvolver-$(VERSION)
 VERSION = 0.9.1
-LDFLAGS += -L$(PREFIX)/$(LIBDIR)
-CPPFLAGS += -O3 -Wall -MMD -MP -DVERSION=\"$(VERSION)\"
-CPPFLAGS += -march=native
-
+CPPFLAGS += -Wall -MMD -MP -DVERSION=\"$(VERSION)\"
+CXXFLAGS = -march=native -O3
+CXX = g++
 
 all:	jconvolver fconvolver makemulti
 
@@ -35,7 +34,7 @@
 JCONVOLVER_O =	jconvolver.o config.o jconfig.o audiofile.o sstring.o jclient.o
 jconvolver:	LDLIBS += -lzita-convolver -lfftw3f -lsndfile -lclthreads -ljack -lpthread -lrt
 jconvolver:	$(JCONVOLVER_O)
-	g++ $(LDFLAGS) -o $@ $(JCONVOLVER_O) $(LDLIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(JCONVOLVER_O) $(LDLIBS) -o $@
 $(JCONVOLVER_O):
 -include $(JCONVOLVER_O:%.o=%.d)
 
@@ -44,7 +43,7 @@
 FCONVOLVER_O =	fconvolver.o config.o fconfig.o audiofile.o sstring.o
 fconvolver:	LDLIBS += -lzita-convolver -lfftw3f -lsndfile -lpthread -lrt
 fconvolver:	$(FCONVOLVER_O)
-	g++ $(LDFLAGS) -o $@ $(FCONVOLVER_O) $(LDLIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(FCONVOLVER_O) $(LDLIBS) -o $@
 $(FCONVOLVER_O):
 -include $(FCONVOLVER_O:%.o=%.d)
 
@@ -53,7 +52,7 @@
 MAKEMULTI_O =	makemulti.o audiofile.o
 makemulti : LDLIBS += -lsndfile -lrt
 makemulti:	$(MAKEMULTI_O)
-	g++ $(LDFLAGS) -o $@ $(MAKEMULTI_O) $(LDLIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) $(MAKEMULTI_O) $(LDLIBS) -o $@
 
 
 install:	all
