diff -uNr wxWidgets-2.7.2.orig/src/unix/fontutil.cpp wxWidgets-2.7.2/src/unix/fontutil.cpp
--- wxWidgets-2.7.2.orig/src/unix/fontutil.cpp	2006-11-07 15:18:20.000000000 +0100
+++ wxWidgets-2.7.2/src/unix/fontutil.cpp	2006-11-10 14:55:57.000000000 +0100
@@ -51,6 +51,7 @@
 #else
     #include "wx/x11/private.h"
     #include "wx/gtk/private/string.h"
+    #include "wx/app.h"
 
     #define wxPANGO_CONV(s) (wxConvUTF8.cWX2MB((s)))
 #endif
