--- wxWidgets-2.6.2/src/common/uri.cpp.orig	2006-02-01 08:34:16.664085500 +0100
+++ wxWidgets-2.6.2/src/common/uri.cpp	2006-02-01 08:27:23.390257500 +0100
@@ -886,7 +886,7 @@
                     op += 3;
             }
 
-            m_path = base.m_path.substr(0, bp - base.m_path.c_str()) +
+            m_path = (const wxString)base.m_path.substr(0, bp - base.m_path.c_str()) +
                     m_path.substr((op - m_path.c_str()), m_path.Length());
         }
     }
