--- wxWidgets-2.5.3/configure.in.orig	2005-02-20 19:38:45.461014000 +0100
+++ wxWidgets-2.5.3/configure.in	2005-02-20 20:58:15.485860176 +0100
@@ -2218,7 +2217,7 @@
             dnl HAVE_STDLIB_H which are already defined and this provokes
             dnl a compiler warning which configure considers as an error...
             AC_MSG_CHECKING(for jpeglib.h)
-            AC_CACHE_VAL(ac_cv_header_jpeglib_h,
+            AC_CACHE_VAL(ac_cv_header_jpeglib_h,[
                 AC_TRY_COMPILE(
                     [
                         #undef HAVE_STDLIB_H
@@ -2230,7 +2229,7 @@
                     ac_cv_header_jpeglib_h=yes,
                     ac_cv_header_jpeglib_h=no
                 )
-            )
+            ])
             AC_MSG_RESULT($ac_cv_header_jpeglib_h)
 
             if test "$ac_cv_header_jpeglib_h" = "yes"; then
