From 3417e2c36779b6eccdaf4e43c56bf48e2db549fc Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Sun, 04 Sep 2011 13:00:13 +0000
Subject: timedate: fix interfaces list

https://bugs.freedesktop.org/show_bug.cgi?id=40583
---
diff --git a/src/timedated.c b/src/timedated.c
index b2a56e7..66e50a6 100644
--- a/src/timedated.c
+++ b/src/timedated.c
@@ -70,7 +70,7 @@
 
 #define INTERFACES_LIST                         \
         BUS_GENERIC_INTERFACES_LIST             \
-        "org.freedesktop.locale1\0"
+        "org.freedesktop.timedate1\0"
 
 const char timedate_interface[] _introspect_("timedate1") = INTERFACE;
 
--
cgit v0.9.0.2-2-gbebe
