diff -aurN openssl-0.9.8x.orig/crypto/cryptlib.h openssl-0.9.8x/crypto/cryptlib.h
--- openssl-0.9.8x.orig/crypto/cryptlib.h	2012-06-03 13:11:20.329957951 +0200
+++ openssl-0.9.8x/crypto/cryptlib.h	2012-06-03 13:15:05.985136873 +0200
@@ -82,7 +82,7 @@
 #ifndef OPENSSL_SYS_VMS
 #define X509_CERT_AREA		OPENSSLDIR
 #define X509_CERT_DIR		OPENSSLDIR "/certs"
-#define X509_CERT_FILE		OPENSSLDIR "/cert.pem"
+#define X509_CERT_FILE		"/etc/certs/ca-certificates.crt"
 #define X509_PRIVATE_DIR	OPENSSLDIR "/private"
 #else
 #define X509_CERT_AREA		"SSLROOT:[000000]"
