--- tk8.5.10/unix/configure.in.orig	2011-06-29 20:07:08.254526698 +0200
+++ tk8.5.10/unix/configure.in	2011-06-29 20:17:19.254547155 +0200
@@ -529,7 +529,7 @@
 	if test "$found_xft" = "no" ; then
 	    found_xft=yes
 	    XFT_CFLAGS=`pkg-config --cflags xft 2>/dev/null` || found_xft="no"
-	    XFT_LIBS=`pkg-config --libs xft 2>/dev/null` || found_xft="no"
+	    XFT_LIBS=`pkg-config --libs xft fontconfig 2>/dev/null` || found_xft="no"
 	fi
 	AC_MSG_RESULT([$found_xft])
 	dnl make sure that compiling against Xft header file doesn't bomb
