diff -urN tk8.5a3.org/unix/Makefile.in tk8.5a3/unix/Makefile.in
--- tk8.5a3.org/unix/Makefile.in	2005-06-04 23:00:25.000000000 +0200
+++ tk8.5a3/unix/Makefile.in	2005-06-17 23:57:52.837126312 +0200
@@ -262,8 +262,8 @@
 # Additional search flags needed to find the various shared libraries
 # at run-time.  The first symbol is for use when creating a binary
 # with cc, and the second is for use when running ld directly.
-CC_SEARCH_FLAGS	= @CC_SEARCH_FLAGS@
-LD_SEARCH_FLAGS	= @LD_SEARCH_FLAGS@
+CC_SEARCH_FLAGS	=
+LD_SEARCH_FLAGS	=
 
 # support for embedded libraries on Darwin / Mac OS X
 DYLIB_INSTALL_DIR	= ${LIB_RUNTIME_DIR}

diff -urN tk8.5a3.org/unix/tcl.m4 tk8.5a3/unix/tcl.m4
--- tk8.5a3.org/unix/tcl.m4	2005-06-04 23:07:14.000000000 +0200
+++ tk8.5a3/unix/tcl.m4	2005-06-17 23:44:37.081099616 +0200
@@ -2042,7 +2042,7 @@
 
     AS_IF([test "${SHARED_BUILD}" = 1 -a "${SHLIB_SUFFIX}" != ""], [
         LIB_SUFFIX=${SHARED_LIB_SUFFIX}
-        MAKE_LIB='${SHLIB_LD} -o [$]@ ${OBJS} ${SHLIB_LD_LIBS} ${TCL_SHLIB_LD_EXTRAS} ${TK_SHLIB_LD_EXTRAS} ${LD_SEARCH_FLAGS}'
+        MAKE_LIB='${SHLIB_LD} -o [$]@ ${OBJS} ${SHLIB_LD_LIBS} ${TCL_SHLIB_LD_EXTRAS} ${TK_SHLIB_LD_EXTRAS}'
         INSTALL_LIB='$(INSTALL_LIBRARY) $(LIB_FILE) "$(LIB_INSTALL_DIR)"/$(LIB_FILE)'
     ], [
         LIB_SUFFIX=${UNSHARED_LIB_SUFFIX}
