--- gcc-4.2.0/libstdc++-v3/configure.orig	2007-06-03 03:02:48.949324693 -0400
+++ gcc-4.2.0/libstdc++-v3/configure	2007-06-03 03:11:31.035566610 -0400
@@ -5865,7 +5865,7 @@
         int j;
         __locale_t        loc;
         __locale_t        loc_dup;
-        loc = __newlocale(1 << LC_ALL, "de_DE", 0);
+        loc = __newlocale(1 << LC_ALL, "C", 0);
         loc_dup = __duplocale(loc);
         i = __strcoll_l(__one, __two, loc);
         j = __strcoll_l(__one, __two, loc_dup);
