diff -urN xinit-1.0.2.orig/Makefile.am xinit-1.0.2/Makefile.am
--- xinit-1.0.2.orig/Makefile.am	2006-08-23 14:00:35.000000000 +0200
+++ xinit-1.0.2/Makefile.am	2006-08-23 14:05:01.000000000 +0200
@@ -19,7 +19,7 @@
 #  TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
 #  PERFORMANCE OF THIS SOFTWARE.
 
-XINITDIR = $(libdir)/X11/xinit
+XINITDIR = /etc/X11/xinit
 
 bin_PROGRAMS = xinit
 bin_SCRIPTS = startx
