--- binutils-2.19.51.0.4/configure.ac~	2009-04-22 01:13:43.033504635 +0200
+++ binutils-2.19.51.0.4/configure.ac	2009-04-22 01:14:26.780226141 +0200
@@ -20,7 +20,6 @@
 ### WARNING: this file contains embedded tabs.  Do not run untabify on this file.
 
 m4_include(config/acx.m4)
-m4_include(config/override.m4)
 m4_include(config/proginstall.m4)
 
 AC_INIT(move-if-change)
@@ -2113,7 +2112,11 @@
 
 # Some systems (e.g., one of the i386-aix systems the gas testers are
 # using) don't handle "\$" correctly, so don't use it here.
-tooldir='${exec_prefix}'/${target_noncanonical}
+if test x${with_tooldir} != x ; then
+	tooldir="${with_tooldir}"
+else
+	tooldir='${exec_prefix}'/${target_noncanonical}
+fi
 build_tooldir=${tooldir}
 
 # Create a .gdbinit file which runs the one in srcdir
