diff -aurN libnotify-0.7.4.orig/configure.ac libnotify-0.7.4/configure.ac
--- libnotify-0.7.4.orig/configure.ac	2012-03-20 16:46:05.680177869 +0100
+++ libnotify-0.7.4/configure.ac	2012-03-20 16:46:36.035798372 +0100
@@ -86,16 +86,12 @@
 LT_PREREQ([2.2])
 LT_INIT
 
-REQ_GTK_VERSION=2.90
 REQ_GLIB_VERSION=2.26.0
 
 pkg_modules="gdk-pixbuf-2.0 glib-2.0 >= $REQ_GLIB_VERSION gio-2.0 >= $REQ_GLIB_VERSION"
 AC_SUBST([pkg_modules])
 PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
 
-tests_modules="gtk+-3.0 >= $REQ_GTK_VERSION"
-PKG_CHECK_MODULES(TESTS, [$tests_modules])
-
 GLIB_GENMARSHAL=`pkg-config --variable=glib_genmarshal glib-2.0`
 AC_SUBST(GLIB_GENMARSHAL)
 
diff -aurN libnotify-0.7.4.orig/Makefile.am libnotify-0.7.4/Makefile.am
--- libnotify-0.7.4.orig/Makefile.am	2012-03-20 16:46:05.678177895 +0100
+++ libnotify-0.7.4/Makefile.am	2012-03-20 16:46:42.508717448 +0100
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-SUBDIRS = libnotify docs tools tests
+SUBDIRS = libnotify docs tools
 
 pcdata_DATA = libnotify.pc
 pcdatadir = $(libdir)/pkgconfig
