diff -urN beecrypt-4.1.0.org/acinclude.m4 beecrypt-4.1.0/acinclude.m4
--- beecrypt-4.1.0.org/acinclude.m4	2004-11-18 13:31:16.000000000 +0100
+++ beecrypt-4.1.0/acinclude.m4	2004-11-18 15:18:11.817798072 +0100
@@ -495,19 +495,19 @@
         # -march=athlon64 degrades performance
         ;;
       athlon*)
-        CFLAGS="$CFLAGS -march=$bc_target_arch"
+        CFLAGS="$CFLAGS -march=$bc_target_cpu"
         ;;
       i586)
-        CFLAGS="$CFLAGS -march=pentium"
+        CFLAGS="$CFLAGS -march=$bc_target_cpu"
         ;;
       i686)
-        CFLAGS="$CFLAGS -march=pentiumpro"
+        CFLAGS="$CFLAGS -march=$bc_target_cpu"
         ;;
       pentium*)
-        CFLAGS="$CFLAGS -march=$bc_target_arch"
+        CFLAGS="$CFLAGS -march=$bc_target_cpu"
         ;;
       powerpc | powerpc64)
-        CFLAGS="$CFLAGS -mcpu=$bc_target_arch"
+        CFLAGS="$CFLAGS -mcpu=$bc_target_cpu"
         ;;
       sparcv8)
         CFLAGS="$CFLAGS -mv8"
@@ -570,7 +570,7 @@
     if test "$bc_cv_c_aggressive_opt" = yes; then
       case $bc_target_cpu in
       athlon*)
-        CXXFLAGS="$CXXFLAGS -mcpu=pentiumpro";
+        CXXFLAGS="$CXXFLAGS -mcpu=$bc_target_cpu";
         ;;
       i586)
         CXXFLAGS="$CXXFLAGS -mcpu=pentium"
@@ -582,7 +582,7 @@
         # no -mcpu=... option on ia64
         ;;
       pentium*)
-        CXXFLAGS="$CXXFLAGS -mcpu=$bc_target_arch"
+        CXXFLAGS="$CXXFLAGS -mcpu=$bc_target_cpu"
         ;;
       esac
       # Architecture-specific optimizations
diff -urN beecrypt-4.1.0.org/configure.ac beecrypt-4.1.0/configure.ac
--- beecrypt-4.1.0.org/configure.ac	2004-11-17 20:05:37.000000000 +0100
+++ beecrypt-4.1.0/configure.ac	2004-11-18 15:19:07.577852788 +0100
@@ -489,7 +489,7 @@
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_ARM"
     ;;
   athlon*)
-    CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I386 -DOPTIMIZE_MMX"
+    CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I686 -DOPTIMIZE_MMX"
     CFLAGS="$CFLAGS -mmmx"
     ;;
   i386)
