--- nspr-4.9/mozilla/nsprpub/configure.in.orig	2012-02-14 20:06:00.000000000 +0100
+++ nspr-4.9/mozilla/nsprpub/configure.in	2012-02-22 19:51:15.304052833 +0100
@@ -596,7 +596,8 @@
 
     AC_MSG_CHECKING([whether the $host compiler ($HOST_CC $HOST_CFLAGS $HOST_LDFLAGS) works])
     AC_TRY_COMPILE([], [return(0);], 
-        [ac_cv_prog_host_cc_works=1 AC_MSG_RESULT([yes])],
+        [ac_cv_prog_host_cc_works=1
+	 AC_MSG_RESULT([yes])],
         AC_MSG_ERROR([installation or configuration problem: $host compiler $HOST_CC cannot create executables.]) )
 
     CC=$_SAVE_CC
@@ -2708,8 +2709,8 @@
     ;;
 *)
     AC_CHECK_LIB(dl, dlopen,
-        AC_CHECK_HEADER(dlfcn.h,
-            OS_LIBS="-ldl $OS_LIBS"))
+        [AC_CHECK_HEADER(dlfcn.h,
+            OS_LIBS="-ldl $OS_LIBS")])
     ;;
 esac
 
@@ -3350,7 +3351,7 @@
 dnl ========================================================
 dnl Generate output files.
 dnl ========================================================
-MAKEFILES="
+m4_define([MAKEFILES], [
     Makefile
     config/Makefile
     config/autoconf.mk
@@ -3380,41 +3381,17 @@
     pr/src/threads/Makefile
     pr/tests/Makefile
     pr/tests/dll/Makefile
-"
-
-if test "$OS_TARGET" = "Linux"; then
-    MAKEFILES="$MAKEFILES
         pkg/linux/Makefile
-    "
-elif test "$OS_TARGET" = "SunOS"; then
-    MAKEFILES="$MAKEFILES
         pkg/solaris/Makefile
         pkg/solaris/SUNWpr/Makefile
         pkg/solaris/SUNWprd/Makefile
-    "
-fi
-
-if test -z "$USE_PTHREADS" && test -z "$USE_BTHREADS"; then
-    MAKEFILES="$MAKEFILES
         pr/src/threads/combined/Makefile
-    "
-elif test -n "$USE_PTHREADS"; then
-    MAKEFILES="$MAKEFILES
         pr/src/pthreads/Makefile
-    "
-elif test -n "$USE_BTHREADS"; then
-    MAKEFILES="$MAKEFILES
         pr/src/bthreads/Makefile
-    "
-fi
-
-if test -n "$USE_CPLUS"; then
-    MAKEFILES="$MAKEFILES
         pr/src/cplus/Makefile
         pr/src/cplus/tests/Makefile
-    "
-fi
+])
 
-echo $MAKEFILES > unallmakefiles
+echo "MAKEFILES" > unallmakefiles
 
-AC_OUTPUT([$MAKEFILES], [chmod +x config/nspr-config])
+AC_OUTPUT(MAKEFILES, [chmod +x config/nspr-config])
