--- autoconf-2.60/lib/autoconf/general.m4.orig	2006-06-16 23:05:41.000000000 +0200
+++ autoconf-2.60/lib/autoconf/general.m4	2006-06-27 23:05:25.877762000 +0200
@@ -2326,7 +2326,7 @@
 AS_IF([dnl eval is necessary to expand ac_cpp.
 dnl Ultrix and Pyramid sh refuse to redirect output of eval, so use subshell.
 (eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD |
-dnl Quote $1 to prevent m4 from eating character classes
+dnl Quote first arg to prevent m4 from eating character classes
   $EGREP "[$1]" >/dev/null 2>&1],
   [$3],
   [$4])dnl
