diff -aurN pygobject-2.15.4.orig/gobject/gobjectmodule.c pygobject-2.15.4/gobject/gobjectmodule.c
--- pygobject-2.15.4.orig/gobject/gobjectmodule.c	2008-09-27 09:04:27.000000000 +0200
+++ pygobject-2.15.4/gobject/gobjectmodule.c	2008-09-27 09:11:12.000000000 +0200
@@ -1809,6 +1809,9 @@
 {
     if (!pyglib_enable_threads())
 	return -1;
+
+    pygobject_api_functions.threads_enabled = TRUE;
+
     return 0;
 }
 
