diff -Nur attr-2.4.32/libattr/Makefile attr-2.4.32.LDFLAGS/libattr/Makefile
--- attr-2.4.32/libattr/Makefile	2006-06-20 08:51:25.000000000 +0200
+++ attr-2.4.32.LDFLAGS/libattr/Makefile	2006-08-02 03:25:39.000000000 +0200
@@ -4,7 +4,7 @@
 
 TOPDIR = ..
 
-LTLDFLAGS += -Wl,--version-script,$(TOPDIR)/exports
+LTLDFLAGS += -Wl,--version-script,$(TOPDIR)/exports $(LLDFLAGS)
 include $(TOPDIR)/include/builddefs
 
 LTLIBRARY = libattr.la
