--- claws-mail-extra-plugins-3.8.1/geolocation_plugin-0.0.8/configure.ac~	2011-12-16 12:20:45.000000000 +0100
+++ claws-mail-extra-plugins-3.8.1/geolocation_plugin-0.0.8/configure.ac	2012-01-02 12:51:12.947113060 +0100
@@ -103,7 +103,9 @@
 	CHAMPLAIN_VERSION=0.6.0], [])
 PKG_CHECK_EXISTS(champlain-gtk-0.8 > 0.8.0,[CHAMPLAIN_MODULE=champlain-gtk-0.8
 	CHAMPLAIN_VERSION=0.8.0], [])
-PKG_CHECK_MODULES(CHAMPLAIN, $CHAMPLAIN_MODULE >= $CHAMPLAIN_VERSION clutter-gtk-0.10)
+PKG_CHECK_EXISTS(champlain-gtk-0.12 > 0.12.0,[CHAMPLAIN_MODULE=champlain-gtk-0.12
+	CHAMPLAIN_VERSION=0.12.0], [])
+PKG_CHECK_MODULES(CHAMPLAIN, $CHAMPLAIN_MODULE >= $CHAMPLAIN_VERSION clutter-gtk-1.0)
 
 AC_SUBST(VERSION)
 AC_SUBST(PLUGINVERSION)
--- claws-mail-extra-plugins-3.8.1/geolocation_plugin-0.0.8/src/geolocation_plugin.c~	2011-09-21 08:19:41.000000000 +0200
+++ claws-mail-extra-plugins-3.8.1/geolocation_plugin-0.0.8/src/geolocation_plugin.c	2012-01-02 12:59:59.397060657 +0100
@@ -276,7 +276,7 @@
   /* create map widget */
   map = gtk_champlain_embed_new();
   view = gtk_champlain_embed_get_view(GTK_CHAMPLAIN_EMBED(map));
-  g_object_set(G_OBJECT(view), "scroll-mode", CHAMPLAIN_SCROLL_MODE_KINETIC, "zoom-level", 8, NULL);
+  g_object_set(G_OBJECT(view), "zoom-level", 8, "kinetic-mode", TRUE, NULL);
 
   /* mail origin marker */
   layer = champlain_layer_new();
