--- gd-2.0.23/gd.h.orig	2004-04-21 23:19:49.000000000 +0200
+++ gd-2.0.23/gd.h	2004-04-21 23:24:10.433748224 +0200
@@ -40,7 +40,7 @@
 /* 2.0.23: more Type 1 fonts */
 #ifndef DEFAULT_FONTPATH
 /* default fontpath for unix systems  - whatever happened to standards ! */
-#define DEFAULT_FONTPATH "/usr/X11R6/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/truetype:/usr/X11R6/lib/X11/fonts/TTF:/usr/share/fonts/TrueType:/usr/share/fonts/truetype:/usr/openwin/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/Type1:/usr/lib/X11/fonts/Type1:/usr/openwin/lib/X11/fonts/Type1"
+#define DEFAULT_FONTPATH "/usr/share/fonts/TTF:/usr/share/fonts/Type1"
 #endif
 #ifndef PATHSEPARATOR
 #define PATHSEPARATOR ":"
