--- qt-everywhere-opensource-src-4.8.0/configure~	2011-12-15 15:34:08.000000000 +0100
+++ qt-everywhere-opensource-src-4.8.0/configure	2011-12-15 15:35:49.522563474 +0100
@@ -4916,6 +4916,11 @@
                 EXTRA_LFLAGS="\$(QMAKE_LFLAGS_RPATH)\"$rpath\" $EXTRA_LFLAGS"
             done
         fi
+
+	if [ -n "$OPTFLAGS" ]; then
+		EXTRA_CFLAGS="$EXTRA_CFLAGS $OPTFLAGS"
+	fi
+
         if [ "$BUILD_ON_MAC" = "yes" ]; then
             echo "export MACOSX_DEPLOYMENT_TARGET = 10.5" >> "$mkfile"
             echo "CARBON_LFLAGS =-framework ApplicationServices" >>"$mkfile"
