diff -urN PackageKit-0.4.4/etc/PackageKit.conf.in PackageKit-0.4.4.new/etc/PackageKit.conf.in
--- PackageKit-0.4.4/etc/PackageKit.conf.in	2009-02-20 17:36:32.000000000 +0100
+++ PackageKit-0.4.4.new/etc/PackageKit.conf.in	2009-02-23 16:28:31.000000000 +0100
@@ -86,7 +86,7 @@
 # NOTE: Don't enable this for backends that are slow doing GetPackages()
 #
 # default=true
-UpdatePackageList=true
+UpdatePackageList=false
 
 # Check for running processes when we update packages
 #
diff -urN PackageKit-0.4.4/etc/Vendor.conf PackageKit-0.4.4.new/etc/Vendor.conf
--- PackageKit-0.4.4/etc/Vendor.conf	2009-02-06 17:01:48.000000000 +0100
+++ PackageKit-0.4.4.new/etc/Vendor.conf	2009-02-23 16:30:10.000000000 +0100
@@ -54,7 +54,7 @@
 # The vendor name shown for the 'vendor' group
 #
 # default=My Vendor
-VendorName=My Vendor
+VendorName=Freddix
 
 # The vendor icon name shown for the 'vendor' group
 #
