diff -burN Python-2.3.3.orig/Demo/cgi/cgi1.py Python-2.3.3/Demo/cgi/cgi1.py
--- Python-2.3.3.orig/Demo/cgi/cgi1.py	2002-10-17 18:33:40.000000000 +0200
+++ Python-2.3.3/Demo/cgi/cgi1.py	2004-03-14 19:18:34.990634848 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python
 
 """CGI test 1 - check server setup."""
 
diff -burN Python-2.3.3.orig/Demo/cgi/cgi2.py Python-2.3.3/Demo/cgi/cgi2.py
--- Python-2.3.3.orig/Demo/cgi/cgi2.py	2002-10-16 23:01:27.000000000 +0200
+++ Python-2.3.3/Demo/cgi/cgi2.py	2004-03-14 19:18:41.993570240 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python
 
 """CGI test 2 - basic use of cgi module."""
 
diff -burN Python-2.3.3.orig/Demo/cgi/cgi3.py Python-2.3.3/Demo/cgi/cgi3.py
--- Python-2.3.3.orig/Demo/cgi/cgi3.py	2002-10-17 17:53:02.000000000 +0200
+++ Python-2.3.3/Demo/cgi/cgi3.py	2004-03-14 19:18:46.984811456 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python
 
 """CGI test 3 (persistent data)."""
 
diff -burN Python-2.3.3.orig/Lib/cgi.py Python-2.3.3/Lib/cgi.py
--- Python-2.3.3.orig/Lib/cgi.py	2003-02-27 21:14:32.000000000 +0100
+++ Python-2.3.3/Lib/cgi.py	2004-03-14 19:18:16.120503544 +0100
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 
 # NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
 # intentionally NOT "/usr/bin/env python".  On many systems
diff -burN Python-2.3.3.orig/Misc/HISTORY Python-2.3.3/Misc/HISTORY
--- Python-2.3.3.orig/Misc/HISTORY	2003-10-20 16:34:46.000000000 +0200
+++ Python-2.3.3/Misc/HISTORY	2004-03-14 19:18:16.148499288 +0100
@@ -7239,9 +7239,9 @@
 were invoked.
 
 - It is now recommended to use ``#!/usr/bin/env python'' instead of
-``#!/usr/local/bin/python'' at the start of executable scripts, except
+``#!/usr/bin/python'' at the start of executable scripts, except
 for CGI scripts.  It has been determined that the use of /usr/bin/env
-is more portable than that of /usr/local/bin/python -- scripts almost
+is more portable than that of /usr/bin/python -- scripts almost
 never have to be edited when the Python interpreter lives in a
 non-standard place.  Note that this doesn't work for CGI scripts since
 the python executable often doesn't live in the HTTP server's default
@@ -10526,7 +10526,7 @@
 Python exits or receives a SIGTERM or SIGHUP signal.
 
 The interpreter is now generally assumed to live in
-/usr/local/bin/python (as opposed to /usr/local/python).  The script
+/usr/bin/python (as opposed to /usr/local/python).  The script
 demo/scripts/fixps.py will update old scripts in place (you can easily
 modify it to do other similar changes).
 
diff -burN Python-2.3.3.orig/Tools/faqwiz/faqw.py Python-2.3.3/Tools/faqwiz/faqw.py
--- Python-2.3.3.orig/Tools/faqwiz/faqw.py	1998-04-04 00:27:04.000000000 +0200
+++ Python-2.3.3/Tools/faqwiz/faqw.py	2004-03-14 19:18:16.150498984 +0100
@@ -1,4 +1,4 @@
-#! /usr/local/bin/python
+#! /usr/bin/python
 
 """FAQ wizard bootstrap."""
 
@@ -10,7 +10,7 @@
 # executable.
 
 # You need to edit the first line and the lines that define FAQDIR and
-# SRCDIR, below: change /usr/local/bin/python to where your Python
+# SRCDIR, below: change /usr/bin/python to where your Python
 # interpreter lives, change the value for FAQDIR to where your FAQ
 # lives, and change the value for SRCDIR to where your faqwiz.py
 # module lives.  The faqconf.py and faqcust.py files live there, too.
diff -burN Python-2.3.3.orig/Tools/scripts/fixps.py Python-2.3.3/Tools/scripts/fixps.py
--- Python-2.3.3.orig/Tools/scripts/fixps.py	2003-05-13 20:14:24.000000000 +0200
+++ Python-2.3.3/Tools/scripts/fixps.py	2004-03-14 19:18:16.151498832 +0100
@@ -15,13 +15,13 @@
             print filename, ': can\'t open :', msg
             continue
         line = f.readline()
-        if not re.match('^#! */usr/local/bin/python', line):
-            print filename, ': not a /usr/local/bin/python script'
+        if not re.match('^#! */usr/bin/python', line):
+            print filename, ': not a /usr/bin/python script'
             f.close()
             continue
         rest = f.read()
         f.close()
-        line = re.sub('/usr/local/bin/python',
+        line = re.sub('/usr/bin/python',
                       '/usr/bin/env python', line)
         print filename, ':', `line`
         f = open(filename, "w")
diff -burN Python-2.3.3.orig/Tools/scripts/parseentities.py Python-2.3.3/Tools/scripts/parseentities.py
--- Python-2.3.3.orig/Tools/scripts/parseentities.py	2001-01-17 09:48:39.000000000 +0100
+++ Python-2.3.3/Tools/scripts/parseentities.py	2004-03-14 19:19:23.258297048 +0100
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python
+#!/usr/bin/python
 """ Utility for parsing HTML entity definitions available from:
 
       http://www.w3.org/ as e.g.
