diff -aurN mozilla-2.0.orig/browser/app/profile/firefox.js mozilla-2.0/browser/app/profile/firefox.js
--- mozilla-2.0.orig/browser/app/profile/firefox.js	2011-03-26 11:55:59.000000000 +0100
+++ mozilla-2.0/browser/app/profile/firefox.js	2011-03-26 12:15:14.000000000 +0100
@@ -137,12 +137,12 @@
 pref("app.update.certs.2.commonName", "aus3.mozilla.org");
 
 // Whether or not app updates are enabled
-pref("app.update.enabled", true);
+pref("app.update.enabled", false);
 
 // This preference turns on app.update.mode and allows automatic download and
 // install to take place. We use a separate boolean toggle for this to make
 // the UI easier to construct.
-pref("app.update.auto", true);
+pref("app.update.auto", false);
 
 // Defines how the Application Update Service notifies the user about updates:
 //
@@ -232,7 +233,7 @@
 
 // Whether or not the application should check at startup each time if it 
 // is the default browser.
-pref("browser.shell.checkDefaultBrowser", true);
+pref("browser.shell.checkDefaultBrowser", false);
 
 // 0 = blank, 1 = home (browser.startup.homepage), 2 = last visited page, 3 = resume previous browser session
 // The behavior of option 3 is detailed at: http://wiki.mozilla.org/Session_Restore
@@ -420,6 +421,8 @@
 //  0: no backups created (and deletes all existing backups)
 pref("browser.bookmarks.max_backups",             10);
 
+pref("intl.locale.matchOS", true);
+
 // Scripts & Windows prefs
 pref("dom.disable_open_during_load",              true);
 pref("javascript.options.showInConsole",          true);
