diff -aurN mozilla-2.0.orig/xulrunner/installer/libxul-embedding.pc.in mozilla-2.0/xulrunner/installer/libxul-embedding.pc.in
--- mozilla-2.0.orig/xulrunner/installer/libxul-embedding.pc.in	2011-03-23 21:11:37.000000000 +0100
+++ mozilla-2.0/xulrunner/installer/libxul-embedding.pc.in	2011-03-23 21:14:29.000000000 +0100
@@ -1,10 +1,11 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 idldir=%idldir%
 
 Name: libxul-embedding
 Description: Static library for version-independent embedding of the Mozilla runtime
 Version: %MOZILLA_VERSION%
-Libs: -L${sdkdir}/lib -lxpcomglue -ldl
-Cflags: -DXPCOM_GLUE -I${includedir} %WCHAR_CFLAGS%
+Libs: -L${libdir}/xulrunner -lxpcomglue -ldl
+Cflags: -DXPCOM_GLUE -I${includedir} -I${prefix}/include/nspr %WCHAR_CFLAGS%
diff -aurN mozilla-2.0.orig/xulrunner/installer/libxul.pc.in mozilla-2.0/xulrunner/installer/libxul.pc.in
--- mozilla-2.0.orig/xulrunner/installer/libxul.pc.in	2011-03-23 21:11:37.000000000 +0100
+++ mozilla-2.0/xulrunner/installer/libxul.pc.in	2011-03-23 21:15:22.000000000 +0100
@@ -1,5 +1,6 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 idldir=%idldir%
 
@@ -7,5 +8,5 @@
 Description: The Mozilla Runtime and Embedding Engine
 Version: %MOZILLA_VERSION%
 Requires: %NSPR_NAME% >= %NSPR_VERSION%
-Libs: -L${sdkdir}/lib %MOZ_XUL_LINK%
-Cflags: -I${includedir} %WCHAR_CFLAGS%
+Libs: -L${libdir}/xulrunner %MOZ_XUL_LINK%
+Cflags: -I${includedir} -I${prefix}/include/nspr %WCHAR_CFLAGS%
diff -aurN mozilla-2.0.orig/xulrunner/installer/Makefile.in mozilla-2.0/xulrunner/installer/Makefile.in
--- mozilla-2.0.orig/xulrunner/installer/Makefile.in	2011-03-23 21:11:37.000000000 +0100
+++ mozilla-2.0/xulrunner/installer/Makefile.in	2011-03-23 21:15:46.000000000 +0100
@@ -121,6 +121,7 @@
 	-e "s|%includedir%|$(includedir)|" \
 	-e "s|%idldir%|$(idldir)|" \
 	-e "s|%sdkdir%|$(sdkdir)|" \
+	-e "s|%libdir%|$(libdir)|" \
 	-e "s|%MOZ_APP_NAME%|$(MOZ_APP_NAME)|" \
 	-e "s|%MOZILLA_VERSION%|$(MOZ_APP_VERSION)|" \
 	-e "s|%WCHAR_CFLAGS%|$(WCHAR_CFLAGS)|" \
diff -aurN mozilla-2.0.orig/xulrunner/installer/mozilla-js.pc.in mozilla-2.0/xulrunner/installer/mozilla-js.pc.in
--- mozilla-2.0.orig/xulrunner/installer/mozilla-js.pc.in	2011-03-23 21:11:37.000000000 +0100
+++ mozilla-2.0/xulrunner/installer/mozilla-js.pc.in	2011-03-23 21:16:34.000000000 +0100
@@ -1,10 +1,11 @@
 prefix=%prefix%
 sdkdir=%sdkdir%
+libdir=%libdir%
 includedir=%includedir%
 
 Name: JavaScript
 Description: The Mozilla JavaScript Library
 Version: %MOZILLA_VERSION%
 Requires: %NSPR_NAME% >= %NSPR_VERSION%
-Libs: -L${sdkdir}/lib %MOZ_JS_LINK%
+Libs: -L${libdir}/xulrunner %MOZ_JS_LINK%
 Cflags: -I${includedir} -DXP_UNIX -DJS_THREADSAFE
