--- busybox-1.17.3/include/libbb.h.orig	2010-10-09 22:57:13.000000000 +0300
+++ busybox-1.17.3/include/libbb.h	2010-11-08 21:21:38.520518694 +0200
@@ -1677,7 +1677,7 @@
 # define VC_2 "/dev/vc/2"
 # define VC_3 "/dev/vc/3"
 # define VC_4 "/dev/vc/4"
-# define VC_5 "/dev/vc/5"
+# define VC_12 "/dev/vc/12"
 # define VC_FORMAT "/dev/vc/%d"
 # define LOOP_FORMAT "/dev/loop/%d"
 # define LOOP_NAMESIZE (sizeof("/dev/loop/") + sizeof(int)*3 + 1)
@@ -1690,7 +1690,7 @@
 # define VC_2 "/dev/tty2"
 # define VC_3 "/dev/tty3"
 # define VC_4 "/dev/tty4"
-# define VC_5 "/dev/tty5"
+# define VC_12 "/dev/tty12"
 # define VC_FORMAT "/dev/tty%d"
 # define LOOP_FORMAT "/dev/loop%d"
 # define LOOP_NAMESIZE (sizeof("/dev/loop") + sizeof(int)*3 + 1)
--- busybox/init/init.c.orig	2008-04-19 04:03:13.000000000 +0000
+++ busybox.o/init/init.c	2008-04-26 17:09:58.075653000 +0000
@@ -64,7 +64,7 @@
 /* Static variables */
 static struct init_action *init_action_list = NULL;
 
-static const char *log_console = VC_5;
+static const char *log_console = VC_12;
 
 enum {
 	L_LOG = 0x1,
