--- module-init-tools-0.9.9.orig/backwards_compat.c
+++ module-init-tools-0.9.9/backwards_compat.c
@@ -26,13 +26,13 @@
 	pid_t pid;
 	char ascii_pid[32];
 	char pathname[strlen(argv[0])+1];
-	char oldname[strlen(progname) + strlen(argv[0]) + sizeof(".old")];
+	char oldname[strlen(progname) + strlen(argv[0]) + sizeof(".modutils")];
 
 	memset(pathname, 0, strlen(argv[0])+1);
 	sep = strrchr(argv[0], '/');
 	if (sep)
 		memcpy(pathname, argv[0], sep - argv[0]+1);
-	sprintf(oldname, "%s%s.old", pathname, progname);
+	sprintf(oldname, "%s%s.modutils", pathname, progname);
 
 	/* Recursion detection: we need an env var since we can't
 	   change argv[0] (as older modutils uses it to determine
--- module-init-tools-0.9.9.orig/depmod.c
+++ module-init-tools-0.9.9/depmod.c
@@ -208,13 +208,13 @@
 {
 	char *sep;
 	char pathname[strlen(argv[0])+1];
-	char oldname[strlen("depmod") + strlen(argv[0]) + sizeof(".old")];
+	char oldname[strlen("depmod") + strlen(argv[0]) + sizeof(".modutils")];
 
 	memset(pathname, 0, strlen(argv[0])+1);
 	sep = strrchr(argv[0], '/');
 	if (sep)
 		memcpy(pathname, argv[0], sep - argv[0]+1);
-	sprintf(oldname, "%s%s.old", pathname, "depmod");
+	sprintf(oldname, "%s%s.modutils", pathname, "depmod");
 
 	/* Recursion detection: we need an env var since we can't
 	   change argv[0] (as older modutils uses it to determine
--- module-init-tools-3.0-pre9.orig/generate-modprobe.conf
+++ module-init-tools-3.0-pre9/generate-modprobe.conf
@@ -17,9 +17,9 @@
     cp $TESTING_MODPROBE_CONF $MODPROBECONF
 elif [ "$STDIN" = "1" ]; then
     cat > $MODPROBECONF
-elif [ -x /sbin/modprobe.old ]; then
+elif [ -x /sbin/modprobe.modutils ]; then
     # In sbin.
-    /sbin/modprobe.old -c > $MODPROBECONF
+    /sbin/modprobe.modutils -c > $MODPROBECONF
 elif modprobe.old -c >/dev/null 2>&1; then
     # Somewhere in path.
     modprobe.old -c > $MODPROBECONF
