--- coreutils-6.10/man/Makefile.am.orig	2008-01-11 11:34:22.000000000 +0100
+++ coreutils-6.10/man/Makefile.am	2008-03-02 02:14:20.884410857 +0100
@@ -59,6 +59,7 @@
 false.1:	$(common_dep)	$(srcdir)/false.x	../src/false.c
 fmt.1:		$(common_dep)	$(srcdir)/fmt.x		../src/fmt.c
 fold.1:		$(common_dep)	$(srcdir)/fold.x	../src/fold.c
+getgid.1:	$(common_dep)	$(srcdir)/getgid.x	../src/getgid.c
 groups.1:	$(common_dep)	$(srcdir)/groups.x	../src/groups.c
 head.1:		$(common_dep)	$(srcdir)/head.x	../src/head.c
 hostid.1:	$(common_dep)	$(srcdir)/hostid.x	../src/hostid.c
diff -Nur coreutils-4.5.3.orig/man/getgid.1 coreutils-4.5.3/man/getgid.1
--- coreutils-4.5.3.orig/man/getgid.1	Thu Jan  1 01:00:00 1970
+++ coreutils-4.5.3/man/getgid.1	Sun Oct 27 21:40:54 2002
@@ -0,0 +1,22 @@
+.TH GETGID "1" "October 2002" "getgid (coreutils) 4.5.3" "User Commands"
+.SH NAME
+getgid \- print ID of given group
+.SH SYNOPSIS
+.B getgid
+\fIGROUPNAME\fR
+.SH DESCRIPTION
+.PP
+Print group ID for GROUPNAME.
+.PP
+Without any OPTION, print usage information.
+.SH AUTHOR
+Written by Artur Frysiak.
+.SH "REPORTING BUGS"
+Report bugs to <feedback@pld-linux.org>.
+.SH COPYRIGHT
+Copyright \(co 1999 Artur Frysiak.
+.br
+This is free software; see the source for copying conditions.  There is NO
+warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+.SH "SEE ALSO"
+.BR id (1)
--- coreutils-4.5.3.orig/man/getgid.x	2007-02-13 17:29:53.000000000 +0100
+++ coreutils-4.5.3.orig/man/getgid.x	2008-08-08 20:54:14.890808765 +0200
@@ -0,0 +1,4 @@
+[NAME]
+getgid \- print ID of given group
+[DESCRIPTION]
+.\" Add any additional description here
diff -Nur coreutils-4.5.3.orig/man/pl/getgid.1 coreutils-4.5.3/man/pl/getgid.1
--- coreutils-4.5.3.orig/man/pl/getgid.1	Thu Jan  1 01:00:00 1970
+++ coreutils-4.5.3/man/pl/getgid.1	Sun Oct 27 21:46:48 2002
@@ -0,0 +1,22 @@
+.TH GETGID "1" "Pa¼dziernik 2002" "getgid (coreutils) 4.5.3" "Polecenia u¿ytkownika"
+.SH NAZWA
+getgid \- wypisuje identyfikator podanej grupy
+.SH SK£ADNIA
+.B getgid
+\fINAZWA-GRUPY\fR
+.SH OPIS
+.PP
+Wypisuje identyfikator grupy NAZWA-GRUPY.
+.PP
+Bez ¿adnych opcji podaje sposób u¿ycia.
+.SH AUTOR
+Napisane przez Artura Frysiaka.
+.SH "ZG£ASZANIE B£ÊDÓW"
+B³êdy proszê zg³aszaæ (w jêz. polskim lub angielskim) pod <feedback@pld-linux.org>.
+.SH COPYRIGHT
+Copyright \(co 1999 Artur Frysiak.
+.br
+This is free software; see the source for copying conditions.  There is NO
+warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+.SH ZOBACZ TAK¯E
+.BR id (1)
--- coreutils-6.10/src/Makefile.am.orig	2008-03-02 02:12:29.418058752 +0100
+++ coreutils-6.10/src/Makefile.am	2008-03-02 02:14:46.205853844 +0100
@@ -69,6 +69,7 @@
   false		\
   fmt		\
   fold		\
+  getgid	\
   ginstall	\
   groups	\
   head		\
diff -Nur coreutils-4.5.3.orig/src/getgid.c coreutils-4.5.3/src/getgid.c
--- coreutils-4.5.3.orig/src/getgid.c	Thu Jan  1 01:00:00 1970
+++ coreutils-4.5.3/src/getgid.c	Sun Oct 27 21:52:01 2002
@@ -0,0 +1,106 @@
+/* getgid - print GID of given group
+   Copyright (C) 1999 Artur Frysiak <wiget@pld-linux.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
+#include <config.h>
+#include <grp.h>
+#include <sys/types.h>
+#include <getopt.h>
+#include "system.h"
+#include "quote.h"
+#include "error.h"
+
+#define PROGRAM_NAME "getgid"
+
+#define AUTHORS proper_name ("Artur Frysiak")
+
+/* The name by which this program was run. */
+const char *program_name;
+
+static struct option const longopts[] =
+{
+  {GETOPT_HELP_OPTION_DECL},
+  {GETOPT_VERSION_OPTION_DECL},
+  {NULL, 0, NULL, 0}
+};
+
+void
+usage (int status)
+{
+  if (status != EXIT_SUCCESS)
+    fprintf (stderr, _("Try `%s --help' for more information.\n"),
+             program_name);
+  else
+    {
+      printf (_("\
+Usage: %s [OPTION]... [GROUPNAME]...\n\
+"),
+              program_name);
+      fputs (_("\
+Prints ID of given group name.\
+\n\
+"), stdout);
+      fputs (HELP_OPTION_DESCRIPTION, stdout);
+      fputs (VERSION_OPTION_DESCRIPTION, stdout);
+      emit_bug_reporting_address ();
+    }
+  exit (status);
+}
+
+int main(int argc, char **argv)
+{
+	struct group 	*gr;
+	int		optc;
+
+	initialize_main (&argc, &argv);
+	program_name = argv[0];
+
+	setlocale (LC_ALL, "");
+	bindtextdomain (PACKAGE, LOCALEDIR);
+	textdomain (PACKAGE);
+
+	atexit (close_stdout);
+
+	while ((optc = getopt_long (argc, argv, "+", longopts, NULL)) != -1) {
+		switch (optc) {
+			case_GETOPT_HELP_CHAR;
+			case_GETOPT_VERSION_CHAR (PROGRAM_NAME, AUTHORS);
+		default:
+			usage (EXIT_FAILURE);
+		}
+	}
+
+	if (argc < optind + 1)
+	{
+		error (0, 0, _("missing operand"));
+		usage (EXIT_FAILURE);
+	}
+
+	if (optind + 2 < argc)
+	{
+		error (0, 0, _("extra operand %s"), quote (argv[optind + 2]));
+		usage (EXIT_FAILURE);
+	}
+
+	gr = getgrnam(argv[optind]);
+	if (gr == NULL) {
+		error (0, 0, _("cannot find group name %s"), quote(argv[optind]));
+		exit(EXIT_FAILURE);
+	} else
+		printf("%lu\n", (unsigned long int) gr->gr_gid);
+	exit(EXIT_SUCCESS);
+}
+
--- coreutils-4.5.3/po/POTFILES.in.orig	Mon Jul  1 23:26:55 2002
+++ coreutils-4.5.3/po/POTFILES.in	Fri Nov  1 02:25:41 2002
@@ -61,6 +61,7 @@
 src/false.c
 src/fmt.c
 src/fold.c
+src/getgid.c
 src/getlimits.c
 src/group-list.c
 src/groups.c
--- coreutils-6.10/po/pl.po.orig	2008-03-02 02:12:29.526064907 +0100
+++ coreutils-6.10/po/pl.po	2008-03-02 02:15:20.139787629 +0100
@@ -4152,6 +4152,15 @@
 msgid "invalid number of columns: %s"
 msgstr "b³êdna liczba kolumn: %s"
 
+#: src/getgid.c:37
+#, c-format
+msgid ""
+"Usage:\n"
+"\t%s groupname\n"
+msgstr ""
+"SkÅ‚adnia:\n"
+"\t%s nazwa-grupy\n"
+
 #. This is a proper name. See the gettext manual, section Names.
 #: src/getlimits.c:29 src/timeout.c:74 src/truncate.c:42
 msgid "Padraig Brady"

