--- coreutils-4.5.3/src/su.c.orig	Sun Oct 27 21:57:03 2002
+++ coreutils-4.5.3/src/su.c	Sun Oct 27 22:15:05 2002
@@ -154,6 +154,15 @@
 # define DEFAULT_ROOT_LOGIN_PATH "/usr/ucb:/bin:/usr/bin:/etc"
 #endif
 
+/* The default paths which get set are both bogus and oddly influenced
+    by <paths.h> and -D on the commands line. Just to be clear, we'll set
+    these explicitly. -ewt */
+#undef DEFAULT_LOGIN_PATH
+#undef DEFAULT_ROOT_LOGIN_PATH
+#define DEFAULT_LOGIN_PATH "/bin:/usr/bin:/usr/local/bin"
+#define DEFAULT_ROOT_LOGIN_PATH \
+    "/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin"
+	
 /* The shell to run if none is given in the user's passwd entry.  */
 #define DEFAULT_SHELL "/bin/sh"
 
