From a803d47c6af1e1c147bbfa342e2b4cbd960185b4 Mon Sep 17 00:00:00 2001
From: Xan Lopez <xan@igalia.com>
Date: Wed, 28 Mar 2012 16:45:17 +0000
Subject: ephy-session: it's not really possible to close while restoring anymore

---
diff --git a/src/ephy-session.c b/src/ephy-session.c
index 7277662..e9c3924 100644
--- a/src/ephy-session.c
+++ b/src/ephy-session.c
@@ -69,7 +69,6 @@ struct _EphySessionPrivate
 	GtkWidget *quit_interact_dialog;
 
 	guint dont_save : 1;
-	guint quit_while_resuming : 1;
 };
 
 #define BOOKMARKS_EDITOR_ID	"BookmarksEditor"
@@ -540,14 +539,6 @@ ephy_session_dispose (GObject *object)
 
 	LOG ("EphySession disposing");
 
-	/* Only remove the crashed session if we're not shutting down while
-	 * the session resume dialogue was still shown!
-	*/
-	if (priv->quit_while_resuming == FALSE)
-	{
-		session_delete (session, SESSION_CRASHED);
-	}
-
 	session_command_queue_clear (session);
 
 	G_OBJECT_CLASS (ephy_session_parent_class)->dispose (object);
@@ -651,8 +642,6 @@ ephy_session_close (EphySession *session)
 	g_object_ref (ephy_shell_get_default ());
 
 	priv->dont_save = TRUE;
-	/* need to set this up here while the dialogue hasn't been killed yet */
-	priv->quit_while_resuming = priv->resume_window != NULL;	
 
 	/* Clear command queue */
 	session_command_queue_clear (session);
--
cgit v0.9.0.2
