--- setup-2.4.11/iana-etc-2.30/Makefile.orig	2006-11-14 18:53:41.000000000 +0100
+++ setup-2.4.11/iana-etc-2.30/Makefile	2007-11-25 15:16:05.825164209 +0100
@@ -65,10 +65,10 @@
 
 protocols: protocol-numbers protocols.gawk
 	$(AWK) --re-interval -f protocols.gawk -v strip=$(STRIP) \
-	    protocol-numbers > protocols
+	    protocol-numbers | unexpand -a > protocols
 
 services: port-numbers services.gawk
-	$(AWK) -f services.gawk -v strip=$(STRIP) port-numbers > services
+	$(AWK) -f services.gawk -v strip=$(STRIP) port-numbers | unexpand -a > services
 
 dist: clean
 	rm -vrf ../iana-etc-`cat VERSION`
--- setup-2.4.11/iana-etc-2.30/port-aliases.orig	2006-11-14 17:01:10.000000000 +0100
+++ setup-2.4.11/iana-etc-2.30/port-aliases	2007-11-25 15:59:09.044373483 +0100
@@ -12,3 +12,44 @@
 
 nicname whois
 
+# PLD additions
+discard sink null
+systat users
+qotd quote
+chargen ttytst source
+smtp mail
+time timserver
+rlp resource
+name nameserver
+http www www-http
+kerberos krb5 kerberos5
+hostname hostnames
+iso-tsap tsap
+cso csnet-ns cso-ns
+pop2 postoffice pop pop-2
+pop3 pop-3
+sunrpc portmapper
+auth tap
+nntp untp
+imap imap2
+snmptrap snmp-trap
+z39.50 z3950 wais
+biff comsat
+who whod
+shell cmd
+printer spooler
+utime unixtime
+router route routed
+courier rpc
+conference chat
+netnews readnews
+uucp uucpd
+kshell krcmd
+new-rwho new-who
+rmonitor rmonitord
+nntps snntp
+kerberos-iv kerberos4
+telnets stelnet
+postgresql postgres
+jms irrd
+sane-port sane
