diff -aurN udisks-1.0.4.orig/data/80-udisks.rules udisks-1.0.4/data/80-udisks.rules
--- udisks-1.0.4.orig/data/80-udisks.rules	2011-09-30 23:51:17.303151701 +0200
+++ udisks-1.0.4/data/80-udisks.rules	2011-09-30 23:56:12.174465302 +0200
@@ -195,9 +195,8 @@
   ENV{UDISKS_PARTITION_TYPE}=="C12A7328-F81F-11D2-BA4B-00A0C93EC93B|21686148-6449-6E6F-744E-656564454649", \
   ENV{UDISKS_PRESENTATION_HIDE}="1"
 
-# APM recovery/tool partitions which are useless on Linux
-ENV{UDISKS_PARTITION_SCHEME}=="apm", \
-  ENV{ID_CDROM}=="?*", ENV{ID_FS_TYPE}=="udf", ENV{ID_FS_LABEL}=="WD_SmartWare", \
+# WD recovery/tool partitions which are useless on Linux
+ENV{ID_CDROM}=="?*", ENV{ID_FS_TYPE}=="udf", ENV{ID_FS_LABEL}=="WD_SmartWare", \
   ENV{UDISKS_PRESENTATION_HIDE}="1"
 
 # recovery partitions (taken from old hal rules)
