diff -urN curl-7.12.1.org/configure.ac curl-7.12.1/configure.ac
--- curl-7.12.1.org/configure.ac	2004-08-13 02:20:37.003335512 +0200
+++ curl-7.12.1/configure.ac	2004-08-13 02:21:07.172749064 +0200
@@ -392,7 +392,7 @@
 main()
 {
  if (socket(AF_INET6, SOCK_STREAM, 0) < 0)
-   exit(1);
+   exit(0);
  else
    exit(0);
 }
diff -aurN curl-7.21.5.orig/configure.ac curl-7.21.5/configure.ac
--- curl-7.21.5.orig/configure.ac	2011-04-22 22:33:30.000000000 +0200
+++ curl-7.21.5/configure.ac	2011-04-22 22:33:53.000000000 +0200
@@ -29,8 +29,8 @@
 CURL_OVERRIDE_AUTOCONF
 
 dnl configure script copyright
-AC_COPYRIGHT([Copyright (c) 1998 - 2011 Daniel Stenberg, <daniel@haxx.se>
-This configure script may be copied, distributed and modified under the
+AC_COPYRIGHT([Copyright (c) 1998 - 2011 Daniel Stenberg, <daniel@haxx.se> \
+This configure script may be copied, distributed and modified under the	\
 terms of the curl license; see COPYING for more details])
 
 AC_CONFIG_SRCDIR([lib/urldata.h])
