diff -aurN jack_capture-0.9.57.orig/Makefile jack_capture-0.9.57/Makefile
--- jack_capture-0.9.57.orig/Makefile	2011-03-05 11:57:10.000000000 +0100
+++ jack_capture-0.9.57/Makefile	2011-03-05 11:59:25.000000000 +0100
@@ -1,6 +1,6 @@
 
 DESTDIR =
-prefix = /usr/local
+prefix = /usr
 bindir = $(prefix)/bin
 
 VERSION=0.9.57
@@ -8,7 +8,7 @@
 CC=gcc
 CPP=g++
 
-OPTIMIZE=-O3 -mtune=native
+OPTIMIZE=$(OPTFLAGS)
 #OPTIMIZE=-O0 -g
 
 COMPILEFLAGS=$(OPTIMIZE) -DVERSION=\"$(VERSION)\" -Wall
@@ -54,11 +54,10 @@
 
 
 jack_capture: setformat.c jack_capture.c vringbuffer.c Makefile das_config.h config_flags
-	$(CC) $(COMPILEFLAGS) jack_capture.c vringbuffer.c -o jack_capture $(LINKFLAGS) `cat config_flags`
-
+	$(CC) jack_capture.c vringbuffer.c $(COMPILEFLAGS) $(LINKFLAGS) `cat config_flags` -o jack_capture
 
 jack_capture_gui2: jack_capture_gui2.cpp
-	$(CPP) $(OPTIMIZE) jack_capture_gui2.cpp `pkg-config --libs --cflags gtk+-2.0` -o jack_capture_gui2
+	$(CPP) jack_capture_gui2.cpp `pkg-config --libs --cflags gtk+-2.0` $(OPTIMIZE) -o jack_capture_gui2
 
 config_flags: Makefile das_config.h
 	cat das_config.h |grep COMPILEFLAGS|sed s/\\/\\/COMPILEFLAGS// >config_flags
