--- libjpeg-turbo-1.1.1/Makefile.am~	2011-05-18 05:51:06.000000000 +0200
+++ libjpeg-turbo-1.1.1/Makefile.am	2011-08-31 21:08:00.029223303 +0200
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = libjpeg.la libturbojpeg.la
 libjpeg_la_LDFLAGS = -version-info ${SO_MAJOR_VERSION}:${SO_MINOR_VERSION} -no-undefined
-libturbojpeg_la_LDFLAGS = -avoid-version -no-undefined
+libturbojpeg_la_LDFLAGS = -no-undefined
 include_HEADERS = jerror.h jmorecfg.h jpeglib.h turbojpeg.h
 nodist_include_HEADERS = jconfig.h
 
