--- gmime-2.6.1.orig/configure.ac	2009-04-09 14:38:18.000000000 +0200
+++ gmime-2.6.1/configure.ac	2009-04-09 15:12:53.710788863 +0200
@@ -387,13 +387,13 @@
 dnl *** Checks for libsocket ***
 dnl ****************************
 LIBSOCKET=""
-AC_CHECK_LIB(socket, getaddrinfo, LIBSOCKET="-lsocket")
+AC_CHECK_FUNC(getaddrinfo,,[AC_CHECK_LIB(socket, getaddrinfo, LIBSOCKET="-lsocket")])
 
 dnl *************************
 dnl *** Checks for libnsl ***
 dnl *************************
 LIBNSL=""
-AC_CHECK_LIB(nsl, getaddrinfo, LIBNSL="-lnsl")
+AC_CHECK_FUNC(getaddrinfo,,[AC_CHECK_LIB(nsl, getaddrinfo, LIBNSL="-lnsl")])
 
 dnl ****************************
 dnl *** Enable Mono bindings ***
