diff -aurN zita-convolver-2.0.0.orig/libs/Makefile zita-convolver-2.0.0/libs/Makefile
--- zita-convolver-2.0.0.orig/libs/Makefile	2009-12-11 20:37:45.000000000 +0100
+++ zita-convolver-2.0.0/libs/Makefile	2009-12-11 20:40:52.000000000 +0100
@@ -17,7 +17,7 @@
 
 # Modify as required.
 #
-PREFIX = /usr/local
+PREFIX = /usr
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 LIBDIR = lib$(SUFFIX)
 
@@ -28,9 +28,9 @@
 DISTDIR = zita-convolver-$(VERSION)
 
 
-CPPFLAGS += -Wall -I. -fPIC -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -O3
+CPPFLAGS += -Wall -I. -fPIC -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS 
 CPPFLAGS += -mmmx -msse -mfpmath=sse -ffast-math -funroll-loops -fpermissive
-CPPFLAGS += -march=i686
+CPPFLAGS += $(OPTFLAGS)
 #CPPFLAGS += -DVECTORIZE 
 
 LDLFAGS += 
@@ -46,17 +46,16 @@
 
 
 $(ZITA-CONVOLVER_MIN):	$(ZITA-CONVOLVER_O)
-	g++ -shared $(LDFLAGS) -Wl,-soname,$(ZITA-CONVOLVER_MAJ) -o $(ZITA-CONVOLVER_MIN) $(ZITA-CONVOLVER_O) $(ZITA-CONVOLVER_DEP)
+	$(CXX) -shared $(OPTFLAGS) $(LDFLAGS) -Wl,-soname,$(ZITA-CONVOLVER_MAJ) -o $(ZITA-CONVOLVER_MIN) $(ZITA-CONVOLVER_O) $(ZITA-CONVOLVER_DEP)
 
 
 install:	$(ZITA-CONVOLVER_MIN)
-	install -d $(PREFIX)/include
-	install -d $(PREFIX)/$(LIBDIR)
-	install -m 644 $(ZITA-CONVOLVER_H) $(PREFIX)/include
-	install -m 755 $(ZITA-CONVOLVER_MIN) $(PREFIX)/$(LIBDIR)
-	ldconfig
-	ln -sf $(ZITA-CONVOLVER_MIN) $(PREFIX)/$(LIBDIR)/$(ZITA-CONVOLVER_SO)
-
+	install -d $(DESTDIR)$(PREFIX)/include
+	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	install -m 644 $(ZITA-CONVOLVER_H) $(DESTDIR)$(PREFIX)/include
+	install -m 755 $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)
+	ln -sf $(ZITA-CONVOLVER_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-CONVOLVER_MAJ)
+	ln -sf $(ZITA-CONVOLVER_MAJ) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-CONVOLVER_SO)
 
 clean:
 	/bin/rm -f *~ *.o *.a *.d *.so.*
